

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Table of Contents: &mdash; CMAP</title>
  

  
  
    <link rel="shortcut icon" href="../../../../_static/CMAP_logo.png"/>
  
  
  

  
  <script type="text/javascript" src="../../../../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../../../" src="../../../../_static/documentation_options.js"></script>
        <script type="text/javascript" src="../../../../_static/jquery.js"></script>
        <script type="text/javascript" src="../../../../_static/underscore.js"></script>
        <script type="text/javascript" src="../../../../_static/doctools.js"></script>
        <script type="text/javascript" src="../../../../_static/language_data.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
        <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true, "ignoreClass": "document", "processClass": "math|output_area"}})</script>
    
    <script type="text/javascript" src="../../../../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../../../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../../../_static/style.css" type="text/css" />
    <link rel="index" title="Index" href="../../../../genindex.html" />
    <link rel="search" title="Search" href="../../../../search.html" />
    <link rel="next" title="opedia (depreciated)" href="../../opedia_tutorials/opedia_tutorials.html" />
    <link rel="prev" title="Presentations" href="../pycmap_presentations.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../../../index.html" class="icon icon-home"> CMAP
          

          
            
            <img src="../../../../_static/CMAP_logo.png" class="logo" alt="Logo"/>
          
          </a>

          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../getting_started/installation/installation.html">Installation</a></li>
</ul>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../../catalog/catalog.html">Data Catalog</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../api_ref.html">Software Packages</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../pycmap_api_ref.html">pycmap API</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="../pycmap_install.html">Installation</a></li>
<li class="toctree-l3"><a class="reference internal" href="../pycmap_data_retrieval.html">Data Retrieval</a></li>
<li class="toctree-l3"><a class="reference internal" href="../pycmap_data_vizualization.html">Data Visualization</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="../pycmap_presentations.html">Presentations</a><ul class="current">
<li class="toctree-l4 current"><a class="current reference internal" href="#">Table of Contents:</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../opedia_tutorials/opedia_tutorials.html">opedia (depreciated)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cmap4r_api/cmap4r_api.html">cmap4r API</a></li>
</ul>
</li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../../faq_and_contributing/datasubmission.html">Data Submission</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../faq_and_contributing/FAQ.html">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../../faq_and_contributing/contact.html">Contact and Contribute</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../../index.html">CMAP</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../../../index.html">Docs</a> &raquo;</li>
        
          <li><a href="../../api_ref.html">Software Packages</a> &raquo;</li>
        
          <li><a href="../pycmap_api_ref.html">pycmap API</a> &raquo;</li>
        
          <li><a href="../pycmap_presentations.html">Presentations</a> &raquo;</li>
        
      <li>Table of Contents:</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput.container,
div.nbinput.container div.prompt,
div.nbinput.container div.input_area,
div.nbinput.container div[class*=highlight],
div.nbinput.container div[class*=highlight] pre,
div.nboutput.container,
div.nboutput.container div.prompt,
div.nboutput.container div.output_area,
div.nboutput.container div[class*=highlight],
div.nboutput.container div[class*=highlight] pre {
    background: none;
    border: none;
    padding: 0 0;
    margin: 0;
    box-shadow: none;
}

/* avoid gaps between output lines */
div.nboutput.container div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput.container,
div.nboutput.container {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput.container,
    div.nboutput.container {
        flex-direction: column;
    }
}

/* input container */
div.nbinput.container {
    padding-top: 5px;
}

/* last container */
div.nblast.container {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput.container div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput.container div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput.container div.prompt,
div.nboutput.container div.prompt {
    min-width: 5ex;
    padding-top: 0.4em;
    padding-right: 0.4em;
    text-align: right;
    flex: 0;
}
@media (max-width: 540px) {
    div.nbinput.container div.prompt,
    div.nboutput.container div.prompt {
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput.container div.prompt.empty {
        padding: 0;
    }
}

/* disable scrollbars on prompts */
div.nbinput.container div.prompt pre,
div.nboutput.container div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput.container div.input_area,
div.nboutput.container div.output_area {
    padding: 0.4em;
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput.container div.input_area,
    div.nboutput.container div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput.container div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    background: #f5f5f5;
}

/* override MathJax center alignment in output cells */
div.nboutput.container div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput.container div.math p {
    text-align: left;
}

/* standard error */
div.nboutput.container div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }

/* Some additional styling taken form the Jupyter notebook CSS */
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
div.rendered_html th {
  font-weight: bold;
}
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}

/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast.container,
.nboutput.nblast.container {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast.container + .nbinput.container {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<p><img alt="image0" src="https://github.com/simonscmap/pycmap/blob/master/docs/figures/CMAP.png?raw=true" /></p>
<p><em>Mohammad Dehghani Ashkezari</em><a class="reference external" href="mailto:mdehghan&#37;&#52;&#48;uw&#46;edu">mdehghan<span>&#64;</span>uw<span>&#46;</span>edu</a></p>
<p><em>Ginger Armbrust</em></p>
<p><em>Raphael Hagen</em></p>
<p><em>Michael Denholtz</em></p>
<div class="section" id="Table-of-Contents:">
<h1>Table of Contents:<a class="headerlink" href="#Table-of-Contents:" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><p><a class="reference external" href="#installation">Installation</a></p></li>
<li><p><a class="reference external" href="#dataRetrieval">Data Retrieval (selected methods)</a></p>
<ul>
<li><p><a class="reference external" href="#api">API</a></p></li>
<li><p><a class="reference external" href="#catalog">Catalog</a></p></li>
<li><p><a class="reference external" href="#searchCatalog">Search Catalog</a></p></li>
<li><p><a class="reference external" href="#cruiseTrajectory">Cruise Trajectory</a></p></li>
<li><p><a class="reference external" href="#spaceTime">Subset by Space-Time</a></p></li>
<li><p><a class="reference external" href="#matchCruise">Colocalize Along Cruise Track</a></p></li>
<li><p><a class="reference external" href="#query">Query</a></p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://cmap.readthedocs.io/en/latest/user_guide/API_ref/pycmap_api/pycmap_data_vizualization.html">Data Visulization</a></p>
<ul>
<li><p><a class="reference external" href="https://cmap.readthedocs.io/en/latest/user_guide/API_ref/pycmap_api/pycmap_histogram.html#histogram">Histogram</a></p></li>
<li><p><a class="reference external" href="https://cmap.readthedocs.io/en/latest/user_guide/API_ref/pycmap_api/pycmap_time_series.html#timeseries">Time Series</a></p></li>
<li><p><a class="reference external" href="https://cmap.readthedocs.io/en/latest/user_guide/API_ref/pycmap_api/pycmap_rm_cp_3d.html#rmcp3d">Regional Map, Contour Plot, 3D Surface Plot</a></p></li>
<li><p><a class="reference external" href="https://cmap.readthedocs.io/en/latest/user_guide/API_ref/pycmap_api/pycmap_section_map_contour.html#sectionmapcontour">Section Map, Section Contour</a></p></li>
<li><p><a class="reference external" href="https://cmap.readthedocs.io/en/latest/user_guide/API_ref/pycmap_api/pycmap_depth_profile.html#depthprofile">Depth Profile</a></p></li>
<li><p><a class="reference external" href="https://cmap.readthedocs.io/en/latest/user_guide/API_ref/pycmap_api/pycmap_cruise_track.html#cruisetrackplot">Cruise Track</a></p></li>
<li><p><a class="reference external" href="https://cmap.readthedocs.io/en/latest/user_guide/API_ref/pycmap_api/pycmap_correlation_matrix_cruise_track.html#corrmatrixcruise">Correlation Matrix Along Cruise Track</a></p></li>
</ul>
</li>
<li><p><a class="reference external" href="#caseStudy">Case Studies</a></p>
<ul>
<li><p><a class="reference external" href="#caseStudy1">Attach Environmental Parameters to the SeaFlow Observations</a></p></li>
<li><p><a class="reference external" href="#caseStudy2">Inter-Annual Variability of Eddy Induced Temperature Anomaly</a></p></li>
</ul>
</li>
</ul>
<center><h1><p>API: Data Retrieval</p>
</h1></center><p>In order to use pycmap, you will need to obtain an API key from SimonsCMAP website: <a class="reference external" href="https://simonscmap.com">https://simonscmap.com</a>.</p>
<div class="section" id="Note:">
<h2>Note:<a class="headerlink" href="#Note:" title="Permalink to this headline">¶</a></h2>
<p>You may install pycmap on cloud-based jupyter notebooks (such as <a class="reference external" href="https://colab.research.google.com/">Colab</a>) by running the following command in a code-block: <code class="docutils literal notranslate"><span class="pre">!pip</span> <span class="pre">install</span> <span class="pre">pycmap</span></code></p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="c1"># !pip install pycmap -q    #uncomment to install pycmap on Colab</span>
<span class="kn">import</span> <span class="nn">pycmap</span>
<span class="n">pycmap</span><span class="o">.</span><span class="n">__version__</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<blockquote>
<div><dl class="field-list simple">
<dt class="field-odd">execution-count</dt>
<dd class="field-odd"><p>2</p>
</dd>
</dl>
<div class="highlight"><pre>
&#39;0.1.2&#39;
</pre></div></div></blockquote>
<p>Table of Contents ## <a class="reference external" href="https://cmap.readthedocs.io/en/latest/user_guide/API_ref/pycmap_api/pycmap_api.html#pycmapapi">API( )</a> To retrieve data, we need to create an instance of the system’s API and pass the API key. It is no necessary to pass the API key every time you run a code locally, because it will be stored locally. The API class has other optional parameters to adjust its behavior. All parameters can be updated persistently at any point in the code.</p>
</div>
</div>
<p>Register at <a class="reference external" href="https://simonscmap.com">https://simonscmap.com</a> and get and API key, if you haven’t already.</p>
<p>Returns a dataframe containing the details of all variables at Simons CMAP database. This method requires no input.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">api</span><span class="o">.</span><span class="n">get_catalog</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Variable</th>
      <th>Table_Name</th>
      <th>Long_Name</th>
      <th>Unit</th>
      <th>Make</th>
      <th>Sensor</th>
      <th>Process_Level</th>
      <th>Study_Domain</th>
      <th>Temporal_Resolution</th>
      <th>Spatial_Resolution</th>
      <th>...</th>
      <th>Comment</th>
      <th>Dataset_Name</th>
      <th>Data_Source</th>
      <th>Distributor</th>
      <th>Dataset_Description</th>
      <th>Acknowledgement</th>
      <th>Dataset_ID</th>
      <th>ID</th>
      <th>Visualize</th>
      <th>Keywords</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>vgosa</td>
      <td>tblAltimetry_REP</td>
      <td>Geostrophic velocity anomalies: meridian compo...</td>
      <td>m/s</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Reprocessed</td>
      <td>Physics</td>
      <td>Daily</td>
      <td>1/4Â° X 1/4Â°</td>
      <td>...</td>
      <td>The geostrophic velocity anomalies are referen...</td>
      <td>Sea Surface Height and Derived Variables (Repr...</td>
      <td>SL-TAC</td>
      <td>http://marine.copernicus.eu/</td>
      <td>The reprocessed altimetry dataset in CMAP was ...</td>
      <td>Data was provided by: E.U. Copernicus Marine S...</td>
      <td>8</td>
      <td>2</td>
      <td>1.0</td>
      <td>daily, geostrophic velocity anomalies, meridia...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>vgos</td>
      <td>tblAltimetry_REP</td>
      <td>Absolute geostrophic velocity: meridian component</td>
      <td>m/s</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Reprocessed</td>
      <td>Physics</td>
      <td>Daily</td>
      <td>1/4Â° X 1/4Â°</td>
      <td>...</td>
      <td>surface_geostrophic_northward_sea_water_velocity</td>
      <td>Sea Surface Height and Derived Variables (Repr...</td>
      <td>SL-TAC</td>
      <td>http://marine.copernicus.eu/</td>
      <td>The reprocessed altimetry dataset in CMAP was ...</td>
      <td>Data was provided by: E.U. Copernicus Marine S...</td>
      <td>8</td>
      <td>6</td>
      <td>1.0</td>
      <td>daily, geostrophic, meridian component, observ...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>sla</td>
      <td>tblAltimetry_REP</td>
      <td>Sea Level Anomaly</td>
      <td>m</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Reprocessed</td>
      <td>Physics</td>
      <td>Daily</td>
      <td>1/4Â° X 1/4Â°</td>
      <td>...</td>
      <td>The sea level anomaly is the sea surface heigh...</td>
      <td>Sea Surface Height and Derived Variables (Repr...</td>
      <td>SL-TAC</td>
      <td>http://marine.copernicus.eu/</td>
      <td>The reprocessed altimetry dataset in CMAP was ...</td>
      <td>Data was provided by: E.U. Copernicus Marine S...</td>
      <td>8</td>
      <td>9</td>
      <td>1.0</td>
      <td>daily, observation, physics, reprocessed, sate...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>adt</td>
      <td>tblAltimetry_REP</td>
      <td>Absolute Dynamic Topography</td>
      <td>m</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Reprocessed</td>
      <td>Physics</td>
      <td>Daily</td>
      <td>1/4Â° X 1/4Â°</td>
      <td>...</td>
      <td>The absolute dynamic topography is the sea sur...</td>
      <td>Sea Surface Height and Derived Variables (Repr...</td>
      <td>SL-TAC</td>
      <td>http://marine.copernicus.eu/</td>
      <td>The reprocessed altimetry dataset in CMAP was ...</td>
      <td>Data was provided by: E.U. Copernicus Marine S...</td>
      <td>8</td>
      <td>10</td>
      <td>1.0</td>
      <td>absolute dynamic topography, adt, daily, obser...</td>
    </tr>
    <tr>
      <th>4</th>
      <td>ugosa</td>
      <td>tblAltimetry_REP</td>
      <td>Geostrophic velocity anomalies: zonal component</td>
      <td>m/s</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Reprocessed</td>
      <td>Physics</td>
      <td>Daily</td>
      <td>1/4Â° X 1/4Â°</td>
      <td>...</td>
      <td>The geostrophic velocity anomalies are referen...</td>
      <td>Sea Surface Height and Derived Variables (Repr...</td>
      <td>SL-TAC</td>
      <td>http://marine.copernicus.eu/</td>
      <td>The reprocessed altimetry dataset in CMAP was ...</td>
      <td>Data was provided by: E.U. Copernicus Marine S...</td>
      <td>8</td>
      <td>11</td>
      <td>1.0</td>
      <td>daily, geostrophic velocity anomalies zonal co...</td>
    </tr>
    <tr>
      <th>5</th>
      <td>ugos</td>
      <td>tblAltimetry_REP</td>
      <td>Absolute geostrophic velocity: zonal component</td>
      <td>m/s</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Reprocessed</td>
      <td>Physics</td>
      <td>Daily</td>
      <td>1/4Â° X 1/4Â°</td>
      <td>...</td>
      <td>surface_geostrophic_eastward_sea_water_velocity</td>
      <td>Sea Surface Height and Derived Variables (Repr...</td>
      <td>SL-TAC</td>
      <td>http://marine.copernicus.eu/</td>
      <td>The reprocessed altimetry dataset in CMAP was ...</td>
      <td>Data was provided by: E.U. Copernicus Marine S...</td>
      <td>8</td>
      <td>12</td>
      <td>1.0</td>
      <td>absolute, component, daily, geostrophic, obser...</td>
    </tr>
    <tr>
      <th>6</th>
      <td>wind_speed_rms</td>
      <td>tblWind_NRT</td>
      <td>Wind speed root mean square</td>
      <td>m/s</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Near-Real-Time</td>
      <td>Physics</td>
      <td>Six Hourly</td>
      <td>1/4Â° X 1/4Â°</td>
      <td>...</td>
      <td>NaN</td>
      <td>Global Ocean Wind L4 6 hourly</td>
      <td>IFREMER CERSAT</td>
      <td>http://marine.copernicus.eu/</td>
      <td>The Near-Real-Time Wind dataset in CMAP is pro...</td>
      <td>http://marine.copernicus.eu/services-portfolio...</td>
      <td>10</td>
      <td>16</td>
      <td>1.0</td>
      <td>daily, near real time, observation, physics, r...</td>
    </tr>
    <tr>
      <th>7</th>
      <td>eastward_wind_rms</td>
      <td>tblWind_NRT</td>
      <td>Eastward wind speed root mean square</td>
      <td>m/s</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Near-Real-Time</td>
      <td>Physics</td>
      <td>Six Hourly</td>
      <td>1/4Â° X 1/4Â°</td>
      <td>...</td>
      <td>NaN</td>
      <td>Global Ocean Wind L4 6 hourly</td>
      <td>IFREMER CERSAT</td>
      <td>http://marine.copernicus.eu/</td>
      <td>The Near-Real-Time Wind dataset in CMAP is pro...</td>
      <td>http://marine.copernicus.eu/services-portfolio...</td>
      <td>10</td>
      <td>17</td>
      <td>1.0</td>
      <td>daily, eastward, eastward_wind_rms, mean, near...</td>
    </tr>
    <tr>
      <th>8</th>
      <td>wind_stress</td>
      <td>tblWind_NRT</td>
      <td>Wind stress</td>
      <td>Pa</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Near-Real-Time</td>
      <td>Physics</td>
      <td>Six Hourly</td>
      <td>1/4Â° X 1/4Â°</td>
      <td>...</td>
      <td>NaN</td>
      <td>Global Ocean Wind L4 6 hourly</td>
      <td>IFREMER CERSAT</td>
      <td>http://marine.copernicus.eu/</td>
      <td>The Near-Real-Time Wind dataset in CMAP is pro...</td>
      <td>http://marine.copernicus.eu/services-portfolio...</td>
      <td>10</td>
      <td>18</td>
      <td>1.0</td>
      <td>daily, near real time, observation, physics, s...</td>
    </tr>
    <tr>
      <th>9</th>
      <td>eastward_wind</td>
      <td>tblWind_NRT</td>
      <td>Eastward wind speed</td>
      <td>m/s</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Near-Real-Time</td>
      <td>Physics</td>
      <td>Six Hourly</td>
      <td>1/4Â° X 1/4Â°</td>
      <td>...</td>
      <td>NaN</td>
      <td>Global Ocean Wind L4 6 hourly</td>
      <td>IFREMER CERSAT</td>
      <td>http://marine.copernicus.eu/</td>
      <td>The Near-Real-Time Wind dataset in CMAP is pro...</td>
      <td>http://marine.copernicus.eu/services-portfolio...</td>
      <td>10</td>
      <td>19</td>
      <td>1.0</td>
      <td>daily, eastward, eastward_wind, near real time...</td>
    </tr>
    <tr>
      <th>10</th>
      <td>surface_downward_eastward_stress</td>
      <td>tblWind_NRT</td>
      <td>Eastward wind stress</td>
      <td>Pa</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Near-Real-Time</td>
      <td>Physics</td>
      <td>Six Hourly</td>
      <td>1/4Â° X 1/4Â°</td>
      <td>...</td>
      <td>NaN</td>
      <td>Global Ocean Wind L4 6 hourly</td>
      <td>IFREMER CERSAT</td>
      <td>http://marine.copernicus.eu/</td>
      <td>The Near-Real-Time Wind dataset in CMAP is pro...</td>
      <td>http://marine.copernicus.eu/services-portfolio...</td>
      <td>10</td>
      <td>20</td>
      <td>1.0</td>
      <td>daily, eastward, near real time, observation, ...</td>
    </tr>
    <tr>
      <th>11</th>
      <td>wind_speed</td>
      <td>tblWind_NRT</td>
      <td>Wind speed</td>
      <td>m/s</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Near-Real-Time</td>
      <td>Physics</td>
      <td>Six Hourly</td>
      <td>1/4Â° X 1/4Â°</td>
      <td>...</td>
      <td>NaN</td>
      <td>Global Ocean Wind L4 6 hourly</td>
      <td>IFREMER CERSAT</td>
      <td>http://marine.copernicus.eu/</td>
      <td>The Near-Real-Time Wind dataset in CMAP is pro...</td>
      <td>http://marine.copernicus.eu/services-portfolio...</td>
      <td>10</td>
      <td>21</td>
      <td>1.0</td>
      <td>daily, near real time, observation, physics, s...</td>
    </tr>
    <tr>
      <th>12</th>
      <td>surface_downward_northward_stress</td>
      <td>tblWind_NRT</td>
      <td>Northward wind stress</td>
      <td>Pa</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Near-Real-Time</td>
      <td>Physics</td>
      <td>Six Hourly</td>
      <td>1/4Â° X 1/4Â°</td>
      <td>...</td>
      <td>NaN</td>
      <td>Global Ocean Wind L4 6 hourly</td>
      <td>IFREMER CERSAT</td>
      <td>http://marine.copernicus.eu/</td>
      <td>The Near-Real-Time Wind dataset in CMAP is pro...</td>
      <td>http://marine.copernicus.eu/services-portfolio...</td>
      <td>10</td>
      <td>22</td>
      <td>1.0</td>
      <td>daily, near real time, northward, observation,...</td>
    </tr>
    <tr>
      <th>13</th>
      <td>northward_wind</td>
      <td>tblWind_NRT</td>
      <td>Northward wind speed</td>
      <td>m/s</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Near-Real-Time</td>
      <td>Physics</td>
      <td>Six Hourly</td>
      <td>1/4Â° X 1/4Â°</td>
      <td>...</td>
      <td>NaN</td>
      <td>Global Ocean Wind L4 6 hourly</td>
      <td>IFREMER CERSAT</td>
      <td>http://marine.copernicus.eu/</td>
      <td>The Near-Real-Time Wind dataset in CMAP is pro...</td>
      <td>http://marine.copernicus.eu/services-portfolio...</td>
      <td>10</td>
      <td>23</td>
      <td>1.0</td>
      <td>daily, near real time, northward, northward_wi...</td>
    </tr>
    <tr>
      <th>14</th>
      <td>northward_wind_rms</td>
      <td>tblWind_NRT</td>
      <td>Northward wind speed root mean square</td>
      <td>m/s</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Near-Real-Time</td>
      <td>Physics</td>
      <td>Six Hourly</td>
      <td>1/4Â° X 1/4Â°</td>
      <td>...</td>
      <td>NaN</td>
      <td>Global Ocean Wind L4 6 hourly</td>
      <td>IFREMER CERSAT</td>
      <td>http://marine.copernicus.eu/</td>
      <td>The Near-Real-Time Wind dataset in CMAP is pro...</td>
      <td>http://marine.copernicus.eu/services-portfolio...</td>
      <td>10</td>
      <td>24</td>
      <td>1.0</td>
      <td>daily, mean, near real time, northward, northw...</td>
    </tr>
    <tr>
      <th>15</th>
      <td>sst</td>
      <td>tblSST_AVHRR_OI_NRT</td>
      <td>Sea Surface Temperature</td>
      <td>C</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Near-Real-Time</td>
      <td>Physics</td>
      <td>Daily</td>
      <td>1/4Â° X 1/4Â°</td>
      <td>...</td>
      <td>Single-sensor Pathfinder 5.0/5.1 AVHRR SSTs us...</td>
      <td>Sea Surface Temperature (Near Real Time)</td>
      <td>National Centers for Environmental Information</td>
      <td>https://podaac.jpl.nasa.gov/</td>
      <td>The Sea Surface Temperature dataset in CMAP is...</td>
      <td>JPL OurOcean Project. 2010. GHRSST Level 4 G1S...</td>
      <td>9</td>
      <td>26</td>
      <td>1.0</td>
      <td>daily, nasa, near real time, observation, oi, ...</td>
    </tr>
    <tr>
      <th>16</th>
      <td>ftle_bw_sla</td>
      <td>tblLCS_REP</td>
      <td>FTLE backward-in-time using geostrophic veloci...</td>
      <td>1/day</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Reprocessed</td>
      <td>Physics</td>
      <td>Daily</td>
      <td>1/25Â° X 1/25Â°</td>
      <td>...</td>
      <td>Passive tracers are advected for 15 days.</td>
      <td>Lagrangian Coherent Structures (Reprocessed)</td>
      <td>Opedia</td>
      <td>Simons Ocean Atlas</td>
      <td>**Reprocessed LCS (Lagrangian Coherent Structu...</td>
      <td>NaN</td>
      <td>1</td>
      <td>34</td>
      <td>1.0</td>
      <td>backward, bw, daily, ftle, ftle_bw_sla, observ...</td>
    </tr>
    <tr>
      <th>17</th>
      <td>disp_bw_sla</td>
      <td>tblLCS_REP</td>
      <td>Displacement backward-in-time using geostrophi...</td>
      <td>deg</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Reprocessed</td>
      <td>Physics</td>
      <td>Daily</td>
      <td>1/25Â° X 1/25Â°</td>
      <td>...</td>
      <td>Passive tracers are advected for 15 days.</td>
      <td>Lagrangian Coherent Structures (Reprocessed)</td>
      <td>Opedia</td>
      <td>Simons Ocean Atlas</td>
      <td>**Reprocessed LCS (Lagrangian Coherent Structu...</td>
      <td>NaN</td>
      <td>1</td>
      <td>35</td>
      <td>1.0</td>
      <td>backward, bw, daily, disp, disp_bw_sla, displa...</td>
    </tr>
    <tr>
      <th>18</th>
      <td>ftle_fw_sla</td>
      <td>tblLCS_REP</td>
      <td>FTLE forward-in-time using geostrophic velocit...</td>
      <td>1/day</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Reprocessed</td>
      <td>Physics</td>
      <td>Daily</td>
      <td>1/25Â° X 1/25Â°</td>
      <td>...</td>
      <td>Passive tracers are advected for 15 days.</td>
      <td>Lagrangian Coherent Structures (Reprocessed)</td>
      <td>Opedia</td>
      <td>Simons Ocean Atlas</td>
      <td>**Reprocessed LCS (Lagrangian Coherent Structu...</td>
      <td>NaN</td>
      <td>1</td>
      <td>36</td>
      <td>1.0</td>
      <td>daily, forward, ftle, ftle_fw_sla, fw, observa...</td>
    </tr>
    <tr>
      <th>19</th>
      <td>disp_fw_sla</td>
      <td>tblLCS_REP</td>
      <td>Displacement forward-in-time using geostrophic...</td>
      <td>deg</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Reprocessed</td>
      <td>Physics</td>
      <td>Daily</td>
      <td>1/25Â° X 1/25Â°</td>
      <td>...</td>
      <td>Passive tracers are advected for 15 days.</td>
      <td>Lagrangian Coherent Structures (Reprocessed)</td>
      <td>Opedia</td>
      <td>Simons Ocean Atlas</td>
      <td>**Reprocessed LCS (Lagrangian Coherent Structu...</td>
      <td>NaN</td>
      <td>1</td>
      <td>37</td>
      <td>1.0</td>
      <td>daily, disp, disp_fw_sla, displacement, forwar...</td>
    </tr>
    <tr>
      <th>20</th>
      <td>Fe</td>
      <td>tblPisces_NRT</td>
      <td>Mole concentration of dissolved iron in sea water</td>
      <td>mmol/m^3</td>
      <td>Model</td>
      <td>Blend</td>
      <td>Forecast</td>
      <td>Biogeochemistry</td>
      <td>Weekly</td>
      <td>1/2Â° X 1/2Â°</td>
      <td>...</td>
      <td>millimoles of Iron per cubic meter</td>
      <td>Mercator-Pisces Biogeochemistry and Weekly For...</td>
      <td>MERCATOR BIOMER4V1R2</td>
      <td>http://marine.copernicus.eu</td>
      <td>"Produced by Mercator Ocean in Toulouse, Franc...</td>
      <td>Data provided by: E.U. Copernicus Marine Servi...</td>
      <td>17</td>
      <td>39</td>
      <td>1.0</td>
      <td>bio, biogeochem, biogeochemistry, biogo, blend...</td>
    </tr>
    <tr>
      <th>21</th>
      <td>PP</td>
      <td>tblPisces_NRT</td>
      <td>Net primary productivity of Carbon per unit vo...</td>
      <td>g/m^3/day</td>
      <td>Model</td>
      <td>Blend</td>
      <td>Forecast</td>
      <td>Biogeochemistry</td>
      <td>Weekly</td>
      <td>1/2Â° X 1/2Â°</td>
      <td>...</td>
      <td>grams of Carbon per cubic meter per day</td>
      <td>Mercator-Pisces Biogeochemistry and Weekly For...</td>
      <td>MERCATOR BIOMER4V1R2</td>
      <td>http://marine.copernicus.eu</td>
      <td>"Produced by Mercator Ocean in Toulouse, Franc...</td>
      <td>Data provided by: E.U. Copernicus Marine Servi...</td>
      <td>17</td>
      <td>40</td>
      <td>1.0</td>
      <td>bio, biogeochem, biogeochemistry, biogo, blend...</td>
    </tr>
    <tr>
      <th>22</th>
      <td>Si</td>
      <td>tblPisces_NRT</td>
      <td>Mole concentration of Silicate in sea water</td>
      <td>umol/L</td>
      <td>Model</td>
      <td>Blend</td>
      <td>Forecast</td>
      <td>Biogeochemistry</td>
      <td>Weekly</td>
      <td>1/2Â° X 1/2Â°</td>
      <td>...</td>
      <td>NaN</td>
      <td>Mercator-Pisces Biogeochemistry and Weekly For...</td>
      <td>MERCATOR BIOMER4V1R2</td>
      <td>http://marine.copernicus.eu</td>
      <td>"Produced by Mercator Ocean in Toulouse, Franc...</td>
      <td>Data provided by: E.U. Copernicus Marine Servi...</td>
      <td>17</td>
      <td>41</td>
      <td>1.0</td>
      <td>bio, biogeochem, biogeochemistry, biogo, blend...</td>
    </tr>
    <tr>
      <th>23</th>
      <td>NO3</td>
      <td>tblPisces_NRT</td>
      <td>Mole concentration of Nitrate in sea water</td>
      <td>mmol/m^3</td>
      <td>Model</td>
      <td>Blend</td>
      <td>Forecast</td>
      <td>Biogeochemistry</td>
      <td>Weekly</td>
      <td>1/2Â° X 1/2Â°</td>
      <td>...</td>
      <td>NaN</td>
      <td>Mercator-Pisces Biogeochemistry and Weekly For...</td>
      <td>MERCATOR BIOMER4V1R2</td>
      <td>http://marine.copernicus.eu</td>
      <td>"Produced by Mercator Ocean in Toulouse, Franc...</td>
      <td>Data provided by: E.U. Copernicus Marine Servi...</td>
      <td>17</td>
      <td>42</td>
      <td>1.0</td>
      <td>bio, biogeochem, biogeochemistry, biogo, blend...</td>
    </tr>
    <tr>
      <th>24</th>
      <td>CHL</td>
      <td>tblPisces_NRT</td>
      <td>Mass concentration of Chlorophyll in sea water</td>
      <td>mg/m^3</td>
      <td>Model</td>
      <td>Blend</td>
      <td>Forecast</td>
      <td>Biogeochemistry</td>
      <td>Weekly</td>
      <td>1/2Â° X 1/2Â°</td>
      <td>...</td>
      <td>NaN</td>
      <td>Mercator-Pisces Biogeochemistry and Weekly For...</td>
      <td>MERCATOR BIOMER4V1R2</td>
      <td>http://marine.copernicus.eu</td>
      <td>"Produced by Mercator Ocean in Toulouse, Franc...</td>
      <td>Data provided by: E.U. Copernicus Marine Servi...</td>
      <td>17</td>
      <td>43</td>
      <td>1.0</td>
      <td>bio, biogeochem, biogeochemistry, biogo, blend...</td>
    </tr>
    <tr>
      <th>25</th>
      <td>PHYC</td>
      <td>tblPisces_NRT</td>
      <td>Mole concentration of Phytoplankton expressed ...</td>
      <td>mmol/m^3</td>
      <td>Model</td>
      <td>Blend</td>
      <td>Forecast</td>
      <td>Biogeochemistry</td>
      <td>Weekly</td>
      <td>1/2Â° X 1/2Â°</td>
      <td>...</td>
      <td>NaN</td>
      <td>Mercator-Pisces Biogeochemistry and Weekly For...</td>
      <td>MERCATOR BIOMER4V1R2</td>
      <td>http://marine.copernicus.eu</td>
      <td>"Produced by Mercator Ocean in Toulouse, Franc...</td>
      <td>Data provided by: E.U. Copernicus Marine Servi...</td>
      <td>17</td>
      <td>44</td>
      <td>1.0</td>
      <td>bio, biogeochem, biogeochemistry, biogo, blend...</td>
    </tr>
    <tr>
      <th>26</th>
      <td>PO4</td>
      <td>tblPisces_NRT</td>
      <td>Mole concentration of Phosphate in sea water</td>
      <td>mmol/m^3</td>
      <td>Model</td>
      <td>Blend</td>
      <td>Forecast</td>
      <td>Biogeochemistry</td>
      <td>Weekly</td>
      <td>1/2Â° X 1/2Â°</td>
      <td>...</td>
      <td>NaN</td>
      <td>Mercator-Pisces Biogeochemistry and Weekly For...</td>
      <td>MERCATOR BIOMER4V1R2</td>
      <td>http://marine.copernicus.eu</td>
      <td>"Produced by Mercator Ocean in Toulouse, Franc...</td>
      <td>Data provided by: E.U. Copernicus Marine Servi...</td>
      <td>17</td>
      <td>45</td>
      <td>1.0</td>
      <td>bio, biogeochem, biogeochemistry, biogo, blend...</td>
    </tr>
    <tr>
      <th>27</th>
      <td>O2</td>
      <td>tblPisces_NRT</td>
      <td>Mole Concentration of dissolved Oxygen in sea ...</td>
      <td>mmol/m^3</td>
      <td>Model</td>
      <td>Blend</td>
      <td>Forecast</td>
      <td>Biogeochemistry</td>
      <td>Weekly</td>
      <td>1/2Â° X 1/2Â°</td>
      <td>...</td>
      <td>NaN</td>
      <td>Mercator-Pisces Biogeochemistry and Weekly For...</td>
      <td>MERCATOR BIOMER4V1R2</td>
      <td>http://marine.copernicus.eu</td>
      <td>"Produced by Mercator Ocean in Toulouse, Franc...</td>
      <td>Data provided by: E.U. Copernicus Marine Servi...</td>
      <td>17</td>
      <td>46</td>
      <td>1.0</td>
      <td>bio, biogeochem, biogeochemistry, biogo, blend...</td>
    </tr>
    <tr>
      <th>28</th>
      <td>DIC_darwin_clim</td>
      <td>tblDarwin_Nutrient_Climatology</td>
      <td>DIC concentration (climatology)</td>
      <td>mmol C/</td>
      <td>Model</td>
      <td>Blend</td>
      <td>Reprocessed</td>
      <td>Biogeochemistry</td>
      <td>Monthly Climatology</td>
      <td>1/2Â° X 1/2Â°</td>
      <td>...</td>
      <td>NaN</td>
      <td>Darwin Biogeochemistry Climatology Model</td>
      <td>http://darwinproject.mit.edu</td>
      <td>http://darwinproject.mit.edu</td>
      <td>This version of the model is modified from Dut...</td>
      <td>Data provided by: http://darwinproject.mit.edu/</td>
      <td>21</td>
      <td>146</td>
      <td>1.0</td>
      <td>bio, biogeochem, biogeochemistry, biogo, blend...</td>
    </tr>
    <tr>
      <th>29</th>
      <td>NH4_darwin_clim</td>
      <td>tblDarwin_Nutrient_Climatology</td>
      <td>NH4 concentration (climatology)</td>
      <td>mmol N/</td>
      <td>Model</td>
      <td>Blend</td>
      <td>Reprocessed</td>
      <td>Biogeochemistry</td>
      <td>Monthly Climatology</td>
      <td>1/2Â° X 1/2Â°</td>
      <td>...</td>
      <td>NaN</td>
      <td>Darwin Biogeochemistry Climatology Model</td>
      <td>http://darwinproject.mit.edu</td>
      <td>http://darwinproject.mit.edu</td>
      <td>This version of the model is modified from Dut...</td>
      <td>Data provided by: http://darwinproject.mit.edu/</td>
      <td>21</td>
      <td>156</td>
      <td>1.0</td>
      <td>ammonium, bio, biogeochem, biogeochemistry, bi...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>1073</th>
      <td>C13_fixation_rate_stdev</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>Standard deviation of 13C fixation rate</td>
      <td>nmol C Lâ1 dâ1</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Standard deviation from biological replicates ...</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2210</td>
      <td>NaN</td>
      <td>13C, 13C fixation, Gradients, Gradients1, MGL2...</td>
    </tr>
    <tr>
      <th>1074</th>
      <td>Gamma_A_nifH_genes_large</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>Gamma-A nifH  (&gt;3 micron)</td>
      <td>gene copies/L</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Gamma-A nifH gene concentrations in seawater (...</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2211</td>
      <td>NaN</td>
      <td>diazotroph, Gamma-A, Gradients, Gradients1, KO...</td>
    </tr>
    <tr>
      <th>1075</th>
      <td>Gamma_A_nifH_genes_small</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>Gamma-A nifH (0.2-3 micron)</td>
      <td>gene copies/L</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Gamma-A nifH gene concentrations in seawater (...</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2212</td>
      <td>NaN</td>
      <td>diazotroph, Gamma-A, Gradients, Gradients1, KO...</td>
    </tr>
    <tr>
      <th>1076</th>
      <td>Het_1_nifH_genes_large</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>Het-1 nifH  (&gt;3 micron)</td>
      <td>gene copies/L</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Het-1 nifH gene concentrations in seawater (&gt;3...</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2213</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients1, Het-1, KOK1...</td>
    </tr>
    <tr>
      <th>1077</th>
      <td>Het_1_nifH_genes_small</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>Het-1 nifH  (0.2-3 micron)</td>
      <td>gene copies/L</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Het-1 nifH gene concentrations in seawater (0....</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2214</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients1, Het-1, KOK1...</td>
    </tr>
    <tr>
      <th>1078</th>
      <td>Het_2_nifH_genes_large</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>Het-2 nifH  (&gt;3 micron)</td>
      <td>gene copies/L</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Het-2 nifH gene concentrations in seawater (&gt;3...</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2215</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients1, Het-2, KOK1...</td>
    </tr>
    <tr>
      <th>1079</th>
      <td>Het_2_nifH_genes_small</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>Het-2 nifH (0.2-3 micron)</td>
      <td>gene copies/L</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Het-2 nifH gene concentrations in seawater (0....</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2216</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients1, Het-2, KOK1...</td>
    </tr>
    <tr>
      <th>1080</th>
      <td>Het_3_nifH_genes_large</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>Het-3 nifH  (&gt;3 micron)</td>
      <td>gene copies/L</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Het-3 nifH gene concentrations in seawater (&gt;3...</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2217</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients1, Het-3, KOK1...</td>
    </tr>
    <tr>
      <th>1081</th>
      <td>Het_3_nifH_genes_small</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>Het-3 nifH  (0.2-3 micron)</td>
      <td>gene copies/L</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Het-3 nifH gene concentrations in seawater (0....</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2218</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients1, Het-3, KOK1...</td>
    </tr>
    <tr>
      <th>1082</th>
      <td>N2_fixation_rate_mean</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>Mean 15N2 fixation rate</td>
      <td>nmol N Lâ1 dâ1</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Mean 15N2 fixation rate measured via 24 hour d...</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2219</td>
      <td>NaN</td>
      <td>15N2, 15N2 fixation, Gradients, Gradients1, MG...</td>
    </tr>
    <tr>
      <th>1083</th>
      <td>N2_fixation_rate_stdev</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>Standard deviation of 15N2 fixation rate</td>
      <td>nmol N Lâ1 dâ1</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Standard deviation from biological replicates ...</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2220</td>
      <td>NaN</td>
      <td>15N2, 15N2 fixation, Gradients, Gradients1, MG...</td>
    </tr>
    <tr>
      <th>1084</th>
      <td>Trichodesmium_nifH_genes_large</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>Trichodesmium nifH (&gt;3 micron)</td>
      <td>gene copies/L</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Trichodesmium nifH gene concentrations in seaw...</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2221</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients1, KOK1606, ni...</td>
    </tr>
    <tr>
      <th>1085</th>
      <td>Trichodesmium_nifH_genes_small</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>Trichodesmium nifH (0.2-3 micron)</td>
      <td>gene copies/L</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Trichodesmium nifH gene concentrations in seaw...</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2222</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients1, KOK1606, ni...</td>
    </tr>
    <tr>
      <th>1086</th>
      <td>UCYN_A1_nifH_genes_large</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>UCYN-A1 nifH (&gt;3 micron)</td>
      <td>gene copies/L</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>UCYN-A1 nifH gene concentrations in seawater (...</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2223</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients1, KOK1606, ni...</td>
    </tr>
    <tr>
      <th>1087</th>
      <td>UCYN_A1_nifH_genes_small</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>UCYN-A1 nifH (0.2-3 micron)</td>
      <td>gene copies/L</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>UCYN-A1 nifH gene concentrations in seawater (...</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2224</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients1, KOK1606, ni...</td>
    </tr>
    <tr>
      <th>1088</th>
      <td>UCYN_A2_A3_nifH_genes_large</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>UCYN-A2/A3 nifH (&gt;3 micron)</td>
      <td>gene copies/L</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>UCYN-A2/A3 nifH gene concentrations in seawate...</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2225</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients1, KOK1606, ni...</td>
    </tr>
    <tr>
      <th>1089</th>
      <td>UCYN_A2_A3_nifH_genes_small</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>UCYN-A2/A3 nifH (0.2-3 micron)</td>
      <td>gene copies/L</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>UCYN-A2/A3 nifH gene concentrations in seawate...</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2226</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients1, KOK1606, ni...</td>
    </tr>
    <tr>
      <th>1090</th>
      <td>UCYN_B_nifH_genes_large</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>UCYN-B nifH (&gt;3 micron)</td>
      <td>gene copies/L</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>UCYN-B nifH gene concentrations in seawater (&gt;...</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2227</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients1, KOK1606, ni...</td>
    </tr>
    <tr>
      <th>1091</th>
      <td>UCYN_B_nifH_genes_small</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>UCYN-B nifH (0.2-3 micron)</td>
      <td>gene copies/L</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>UCYN-B nifH gene concentrations in seawater (0...</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2228</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients1, KOK1606, ni...</td>
    </tr>
    <tr>
      <th>1092</th>
      <td>UCYN_C_nifH_genes_large</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>UCYN-C nifH (&gt;3 micron)</td>
      <td>gene copies/L</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>UCYN-C nifH gene concentrations in seawater (&gt;...</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2229</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients1, KOK1606, ni...</td>
    </tr>
    <tr>
      <th>1093</th>
      <td>UCYN_C_nifH_genes_small</td>
      <td>tblKOK1606_Gradients1_Diazotroph</td>
      <td>UCYN-C nifH (0.2-3 micron)</td>
      <td>gene copies/L</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>UCYN-C nifH gene concentrations in seawater (0...</td>
      <td>KOK1606 Diazotroph Commuinity Composition, 13C...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>139</td>
      <td>2230</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients1, KOK1606, ni...</td>
    </tr>
    <tr>
      <th>1094</th>
      <td>UCYN_A1_nifH_genes_mean</td>
      <td>tblMGL1704_Gradients2_Diazotroph</td>
      <td>UCYN-A1 nifH (&gt;0.2 micron)</td>
      <td>UCYN-A1 nifH gene copies Lâ1</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Mean UCYN-A1 nifH gene concentrations in seawa...</td>
      <td>MGL1704 Diazotroph Community Composition, 13C ...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>141</td>
      <td>2231</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients2, MGL1704, ni...</td>
    </tr>
    <tr>
      <th>1095</th>
      <td>UCYN_A1_nifH_genes_stdev</td>
      <td>tblMGL1704_Gradients2_Diazotroph</td>
      <td>UCYN-A1 nifH standard deviation (&gt;0.2 micron)</td>
      <td>UCYN-A1 nifH gene copies Lâ1</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Standard deviation from biological replicates ...</td>
      <td>MGL1704 Diazotroph Community Composition, 13C ...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>141</td>
      <td>2232</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients2, MGL1704, ni...</td>
    </tr>
    <tr>
      <th>1096</th>
      <td>N2_fixation_rate_mean</td>
      <td>tblMGL1704_Gradients2_Diazotroph</td>
      <td>Mean 15N2 fixation rate</td>
      <td>nmol N Lâ1 dâ1</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Mean 15N2 fixation rate measured via 24 hour d...</td>
      <td>MGL1704 Diazotroph Community Composition, 13C ...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>141</td>
      <td>2233</td>
      <td>NaN</td>
      <td>15N2, 15N2 fixation, Gradients, Gradients2, MG...</td>
    </tr>
    <tr>
      <th>1097</th>
      <td>N2_fixation_rate_stdev</td>
      <td>tblMGL1704_Gradients2_Diazotroph</td>
      <td>Standard deviation of 15N2 fixation rate</td>
      <td>nmol N Lâ1 dâ1</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Standard deviation from biological replicates ...</td>
      <td>MGL1704 Diazotroph Community Composition, 13C ...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>141</td>
      <td>2234</td>
      <td>NaN</td>
      <td>15N2, 15N2 fixation, Gradients, Gradients2, MG...</td>
    </tr>
    <tr>
      <th>1098</th>
      <td>C13_fixation_rate_mean</td>
      <td>tblMGL1704_Gradients2_Diazotroph</td>
      <td>Mean 13C fixation rate</td>
      <td>nmol C Lâ1 dâ1</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Mean 13C fixation rate measured via 24 hour de...</td>
      <td>MGL1704 Diazotroph Community Composition, 13C ...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>141</td>
      <td>2235</td>
      <td>NaN</td>
      <td>13C, 13C fixation, Gradients, Gradients2, MGL1...</td>
    </tr>
    <tr>
      <th>1099</th>
      <td>C13_fixation_rate_stdev</td>
      <td>tblMGL1704_Gradients2_Diazotroph</td>
      <td>Standard deviation of 13C fixation rate</td>
      <td>nmol C Lâ1 dâ1</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Standard deviation from biological replicates ...</td>
      <td>MGL1704 Diazotroph Community Composition, 13C ...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>141</td>
      <td>2236</td>
      <td>NaN</td>
      <td>13C, 13C fixation, Gradients, Gradients2, MGL1...</td>
    </tr>
    <tr>
      <th>1100</th>
      <td>Trichodesmium_nifH_genes_mean</td>
      <td>tblMGL1704_Gradients2_Diazotroph</td>
      <td>Trichodesmium nifH (&gt;0.2 micron)</td>
      <td>Trichodesmium nifH gene copies Lâ1</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Mean Trichodesmium nifH gene concentrations in...</td>
      <td>MGL1704 Diazotroph Community Composition, 13C ...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>141</td>
      <td>2237</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients2, MGL1704, ni...</td>
    </tr>
    <tr>
      <th>1101</th>
      <td>Trichodesmium_nifH_genes_stdev</td>
      <td>tblMGL1704_Gradients2_Diazotroph</td>
      <td>Trichodesmium nifH standard deviation (&gt;0.2 mi...</td>
      <td>Trichodesmium nifH gene copies Lâ1</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>Standard deviation from biological replicates ...</td>
      <td>MGL1704 Diazotroph Community Composition, 13C ...</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>Zehr lab, UC Santa Cruz</td>
      <td>This study investigates the diversity, abundan...</td>
      <td>NaN</td>
      <td>141</td>
      <td>2238</td>
      <td>NaN</td>
      <td>diazotroph, Gradients, Gradients2, MGL1704, ni...</td>
    </tr>
    <tr>
      <th>1102</th>
      <td>PAR</td>
      <td>tblModis_PAR</td>
      <td>Photosynthetically Available Radiation</td>
      <td>einstein m^-2 day^-1</td>
      <td>Observation</td>
      <td>Satellite</td>
      <td>Reprocessed</td>
      <td>Physics</td>
      <td>Three Days</td>
      <td>9km X 9km</td>
      <td>...</td>
      <td>NaN</td>
      <td>MODIS Photosynthetically Available Radiation 8...</td>
      <td>Ocean Biology Processing Group at NASA GSFC</td>
      <td>Ocean Biology Processing Group at NASA GSFC</td>
      <td>The MODIS PAR dataset in CMAP is satellite der...</td>
      <td>NaN</td>
      <td>143</td>
      <td>2240</td>
      <td>NaN</td>
      <td>Aqua, biogeochemistry, biology, Moderate Resol...</td>
    </tr>
  </tbody>
</table>
<p>1103 rows × 36 columns</p>
</div></div>
</div>
<p>Returns a dataframe containing a subset of Simons CMAP catalog of variables.</p>
<p>All variables at Simons CMAP catalog are annotated with a collection of semantically related keywords. This method takes the passed keywords and returns all of the variables annotated with similar keywords. The passed keywords should be separated by blank space. The search result is not sensitive to the order of keywords and is not case sensitive. The passed keywords can provide any ‘hint’ associated with the target variables. Below are a few examples:</p>
<ul class="simple">
<li><p>the exact variable name (e.g. NO3), or its linguistic term (Nitrate)</p></li>
<li><p>methodology (model, satellite …), instrument (CTD, seaflow), or disciplines (physics, biology …) * the cruise official name (e.g. KOK1606), or unofficial cruise name (Falkor)</p></li>
<li><p>the name of data producer (e.g Penny Chisholm) or institution name (MIT)</p></li>
</ul>
<p>If you searched for a variable with semantically-related-keywords and did not get the correct results, please let us know. We can update the keywords at any point.</p>
</div>
<div class="section" id="Example:">
<h2>Example:<a class="headerlink" href="#Example:" title="Permalink to this headline">¶</a></h2>
<p>Returns a list of Nitrite measurements during the Falkor cruise, if exists.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">api</span><span class="o">.</span><span class="n">search_catalog</span><span class="p">(</span><span class="s1">&#39;nitrite falkor&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Variable</th>
      <th>Table_Name</th>
      <th>Long_Name</th>
      <th>Unit</th>
      <th>Make</th>
      <th>Sensor</th>
      <th>Process_Level</th>
      <th>Study_Domain</th>
      <th>Temporal_Resolution</th>
      <th>Spatial_Resolution</th>
      <th>...</th>
      <th>Comment</th>
      <th>Dataset_Name</th>
      <th>Data_Source</th>
      <th>Distributor</th>
      <th>Dataset_Description</th>
      <th>Acknowledgement</th>
      <th>Dataset_ID</th>
      <th>ID</th>
      <th>Visualize</th>
      <th>Keywords</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>NO3_NO2</td>
      <td>tblFalkor_2018</td>
      <td>Nitrate plus Nitrite</td>
      <td>umol/kg</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biogeochemistry</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>NaN</td>
      <td>2018 SCOPE Falkor Cruise, Water Column Data</td>
      <td>University of Hawaii, Manoa</td>
      <td>University of Hawaii, Manoa</td>
      <td>The 2018 SCOPE-Falkor Cruise was built on the ...</td>
      <td>NaN</td>
      <td>99</td>
      <td>1404</td>
      <td>1</td>
      <td>Autoanalyzer, Core Data, cruise, FK180310, Nit...</td>
    </tr>
  </tbody>
</table>
<p>1 rows × 36 columns</p>
</div></div>
</div>
<p>Returns a dataframe containing the trajectory of the specified cruise.</p>
</div>
<div class="section" id="Example:">
<h2>Example:<a class="headerlink" href="#Example:" title="Permalink to this headline">¶</a></h2>
<p>Returns the meso_scope cruise trajectory. The example below passes ‘scope’ as cruise name. All cruises that have the term ‘scope’ in their name are returned and asks for more specific name.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">api</span><span class="o">.</span><span class="n">cruise_trajectory</span><span class="p">(</span><span class="s1">&#39;scope&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>

More than one cruise found. Please provide a more specific cruise name:
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
     ID       Nickname        Name               Ship_Name  \
0   576        SCOPE_1      KM1427          R/V Kilo Moana
1   577        SCOPE_2      KM1502          R/V Kilo Moana
2   578        SCOPE_3      KM1508          R/V Kilo Moana
3   579        SCOPE_4      KM1510          R/V Kilo Moana
4   580        SCOPE_5      KM1512          R/V Kilo Moana
5   582        SCOPE_9     KOK1512  R/V Kaimikai O Kanaloa
6   583       SCOPE_10     KOK1515  R/V Kaimikai O Kanaloa
7   584       SCOPE_11      KM1518          R/V Kilo Moana
8   585       SCOPE_12      KM1601          R/V Kilo Moana
9   586       SCOPE_13      KM1602          R/V Kilo Moana
10  587       SCOPE_14      KM1603          R/V Kilo Moana
11  588       SCOPE_15     KOK1604  R/V Kaimikai O Kanaloa
12  590       SCOPE_17     KOK1607  R/V Kaimikai O Kanaloa
13  591       SCOPE_18     KOK1608  R/V Kaimikai O Kanaloa
14  592       SCOPE_19     KOK1609  R/V Kaimikai O Kanaloa
15  595     MESO_SCOPE      KM1709          R/V Kilo Moana
16  601  SCOPE_Falkor1  FK180310-1              R/V Falkor
17  602  SCOPE_Falkor2  FK180310-2              R/V Falkor

                  Start_Time                  End_Time  Lat_Min  Lat_Max  \
0   2014-12-08T22:53:34.000Z  2014-12-12T02:33:13.000Z  21.2229  22.8723
1   2015-03-21T04:52:20.000Z  2015-03-29T05:21:56.000Z  21.9813  46.1307
2   2015-05-22T22:53:01.000Z  2015-05-26T16:43:23.000Z  21.2448  22.8688
3   2015-06-18T21:41:45.000Z  2015-06-21T13:04:57.000Z  21.3369  22.8548
4   2015-07-18T22:41:36.000Z  2015-07-22T08:29:56.000Z  21.3415  22.8373
5   2015-09-27T03:12:44.000Z  2015-09-27T08:46:07.000Z  21.2475  21.3551
6   2015-10-13T02:52:33.000Z  2015-10-16T09:40:22.000Z  21.2388  22.8257
7   2015-11-12T23:09:39.000Z  2015-11-16T04:03:21.000Z  21.3429  22.7981
8   2016-01-11T19:51:35.000Z  2016-01-15T07:12:34.000Z  21.2414  22.8480
9   2016-02-08T22:37:47.000Z  2016-02-12T07:51:39.000Z  21.3434  22.8803
10  2016-03-08T03:47:24.000Z  2016-03-10T02:12:13.000Z  21.8343  22.7524
11  2016-04-13T14:13:48.000Z  2016-04-17T01:36:58.000Z  21.5715  22.7715
12  2016-05-12T16:58:49.000Z  2016-05-14T06:50:34.000Z  22.8092  24.4873
13  2016-07-10T20:25:23.000Z  2016-07-14T09:10:17.000Z  21.2511  22.7748
14  2016-08-08T19:33:41.000Z  2016-08-12T08:31:17.000Z  21.7156  22.8043
15  2017-06-26T22:30:09.000Z  2017-07-15T05:09:24.000Z  21.3120  27.9520
16  2018-03-11T19:43:14.000Z  2018-03-23T06:18:32.000Z  20.9801  22.7207
17  2018-03-28T02:26:35.000Z  2018-04-10T02:18:12.000Z  21.1674  24.7104

     Lon_Min   Lon_Max                   Chief_Name
0  -158.3345 -157.8655                Samuel Wilson
1  -156.6921 -124.3023                          NaN
2  -158.3698 -157.8876                Daniel Sadler
3  -158.3355 -157.8908                Susan Curless
4  -158.3676 -157.9158                Daniel Sadler
5  -158.2810 -157.8689                Susan Curless
6  -158.3683 -157.9239              R. Walter Deppe
7  -158.3365 -157.8771  Fernando Santiago-Mandujano
8  -158.3638 -157.8920                Susan Curless
9  -158.3648 -157.4255                Tara Clemente
10 -158.2635 -157.9813                Susan Curless
11 -158.3981 -157.9014                Margo Edwards
12 -157.3852 -155.9214             Benedetto Barone
13 -158.3658 -157.9082               Matthew Church
14 -158.3634 -157.8842                  Terry Kerby
15 -160.3403 -157.2829             Benedetto Barone
16 -158.3520 -155.0850                Samuel Wilson
17 -160.9810 -156.2686                Samuel Wilson
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
An exception has occurred, use %tb to see the full traceback.

<span class="ansi-red-fg">SystemExit</span><span class="ansi-red-fg">:</span> 1

</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<blockquote>
<div><dl class="field-list simple">
<dt class="field-odd">class</dt>
<dd class="field-odd"><p>stderr</p>
</dd>
</dl>
<div class="highlight"><pre>
/home/nrhagen/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3304: UserWarning: To exit: use &#39;exit&#39;, &#39;quit&#39;, or Ctrl-D.
  warn(&#34;To exit: use &#39;exit&#39;, &#39;quit&#39;, or Ctrl-D.&#34;, stacklevel=1)
</pre></div></div></blockquote>
<p>Table of Contents ## <a class="reference external" href="https://cmap.readthedocs.io/en/latest/user_guide/API_ref/pycmap_api/pycmap_cruise_variables.html#cruisevars">cruise_variables(cruiseName)</a></p>
</div>
</div>
<p>Returns a dataframe containing all registered variables (at Simons CMAP) during the specified cruise.</p>
</div>
<div class="section" id="Example:">
<h2>Example:<a class="headerlink" href="#Example:" title="Permalink to this headline">¶</a></h2>
<p>Returns a list of measured variables during the <em>Diel</em> cruise (KM1513).</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">api</span><span class="o">.</span><span class="n">cruise_variables</span><span class="p">(</span><span class="s1">&#39;diel&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Variable</th>
      <th>Table_Name</th>
      <th>Long_Name</th>
      <th>Unit</th>
      <th>Make</th>
      <th>Sensor</th>
      <th>Process_Level</th>
      <th>Study_Domain</th>
      <th>Temporal_Resolution</th>
      <th>Spatial_Resolution</th>
      <th>...</th>
      <th>Variable_Min</th>
      <th>Variable_Max</th>
      <th>Comment</th>
      <th>Dataset_Name</th>
      <th>Data_Source</th>
      <th>Distributor</th>
      <th>Dataset_Description</th>
      <th>Dataset_ID</th>
      <th>ID</th>
      <th>Keywords</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>PAR</td>
      <td>tblCruise_PAR</td>
      <td>Photosynthetically Active Radiation</td>
      <td>Âµmol/m^2/s</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Chemistry+Biology+Biogeochemistry</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>temperature data comes from multiple sources a...</td>
      <td>Cruise PAR Measurements for Cruises in CMAP</td>
      <td>multiple sources - see tblDataset_References</td>
      <td>Multiple Distributors</td>
      <td>Cruise PAR is a aggregation of multiple in-sit...</td>
      <td>NaN</td>
      <td>137</td>
      <td>1982</td>
    </tr>
    <tr>
      <th>1</th>
      <td>sra_experiment</td>
      <td>tblHL2A_diel_metagenomics</td>
      <td>NCBI SRA experiment name</td>
      <td>NaN</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Genomics</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Accecssio n numbers to the metagenomic data at...</td>
      <td>Diel bacterial, archaeal, and phage metagenome...</td>
      <td>DeLong lab, University of Hawaii, Manoa</td>
      <td>University of Hawaii, Manoa</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>110</td>
      <td>1649</td>
    </tr>
    <tr>
      <th>2</th>
      <td>sra_run</td>
      <td>tblHL2A_diel_metagenomics</td>
      <td>NCBI SRA sequence runs</td>
      <td>NaN</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Genomics</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Accecssio n numbers to the metagenomic data at...</td>
      <td>Diel bacterial, archaeal, and phage metagenome...</td>
      <td>DeLong lab, University of Hawaii, Manoa</td>
      <td>University of Hawaii, Manoa</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>110</td>
      <td>1650</td>
    </tr>
    <tr>
      <th>3</th>
      <td>sra_bioproject</td>
      <td>tblHL2A_diel_metagenomics</td>
      <td>NCBI SRA  Bioproject number</td>
      <td>NaN</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Genomics</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Accecssio n numbers to the metagenomic data at...</td>
      <td>Diel bacterial, archaeal, and phage metagenome...</td>
      <td>DeLong lab, University of Hawaii, Manoa</td>
      <td>University of Hawaii, Manoa</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>110</td>
      <td>1651</td>
    </tr>
    <tr>
      <th>4</th>
      <td>sequencing_method</td>
      <td>tblHL2A_diel_metagenomics</td>
      <td>DNA sequencing method</td>
      <td>NaN</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Genomics</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Diel bacterial, archaeal, and phage metagenome...</td>
      <td>DeLong lab, University of Hawaii, Manoa</td>
      <td>University of Hawaii, Manoa</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>110</td>
      <td>1658</td>
    </tr>
    <tr>
      <th>5</th>
      <td>filter_type</td>
      <td>tblHL2A_diel_metagenomics</td>
      <td>filter_type</td>
      <td>NaN</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Diel bacterial, archaeal, and phage metagenome...</td>
      <td>DeLong lab, University of Hawaii, Manoa</td>
      <td>University of Hawaii, Manoa</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>110</td>
      <td>1652</td>
    </tr>
    <tr>
      <th>6</th>
      <td>filter_max</td>
      <td>tblHL2A_diel_metagenomics</td>
      <td>Filter Max</td>
      <td>Âµm</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.200000</td>
      <td>0.200000</td>
      <td>NaN</td>
      <td>Diel bacterial, archaeal, and phage metagenome...</td>
      <td>DeLong lab, University of Hawaii, Manoa</td>
      <td>University of Hawaii, Manoa</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>110</td>
      <td>1653</td>
    </tr>
    <tr>
      <th>7</th>
      <td>filter_min</td>
      <td>tblHL2A_diel_metagenomics</td>
      <td>Filter Min</td>
      <td>Âµm</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.030000</td>
      <td>0.200000</td>
      <td>NaN</td>
      <td>Diel bacterial, archaeal, and phage metagenome...</td>
      <td>DeLong lab, University of Hawaii, Manoa</td>
      <td>University of Hawaii, Manoa</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>110</td>
      <td>1654</td>
    </tr>
    <tr>
      <th>8</th>
      <td>library_kit</td>
      <td>tblHL2A_diel_metagenomics</td>
      <td>Sequencing Library Kit</td>
      <td>NaN</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Genomics</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Diel bacterial, archaeal, and phage metagenome...</td>
      <td>DeLong lab, University of Hawaii, Manoa</td>
      <td>University of Hawaii, Manoa</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>110</td>
      <td>1655</td>
    </tr>
    <tr>
      <th>9</th>
      <td>sequence_type</td>
      <td>tblHL2A_diel_metagenomics</td>
      <td>DNA Sequence Type</td>
      <td>NaN</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Genomics</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Diel bacterial, archaeal, and phage metagenome...</td>
      <td>DeLong lab, University of Hawaii, Manoa</td>
      <td>University of Hawaii, Manoa</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>110</td>
      <td>1656</td>
    </tr>
    <tr>
      <th>10</th>
      <td>DNA_RNA_prep_method</td>
      <td>tblHL2A_diel_metagenomics</td>
      <td>DNA preparation method/kit</td>
      <td>NaN</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Genomics</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Diel bacterial, archaeal, and phage metagenome...</td>
      <td>DeLong lab, University of Hawaii, Manoa</td>
      <td>University of Hawaii, Manoa</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>110</td>
      <td>1657</td>
    </tr>
    <tr>
      <th>11</th>
      <td>cruise</td>
      <td>tblSeaFlow</td>
      <td>Cruise Name</td>
      <td>NaN</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1184</td>
    </tr>
    <tr>
      <th>12</th>
      <td>prochloro_abundance</td>
      <td>tblSeaFlow</td>
      <td>Prochlorococcus Abundance</td>
      <td>cells/Î¼L</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.052346</td>
      <td>247.554453</td>
      <td>cell abundance, see https://github.com/armbrus...</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1185</td>
    </tr>
    <tr>
      <th>13</th>
      <td>prochloro_diameter</td>
      <td>tblSeaFlow</td>
      <td>Prochlorococcus Cell Diameter</td>
      <td>micrometer</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.316687</td>
      <td>0.997106</td>
      <td>middle value of mean cell diameter based on Mi...</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1186</td>
    </tr>
    <tr>
      <th>14</th>
      <td>prochloro_carbon_content</td>
      <td>tblSeaFlow</td>
      <td>Prochlorococcus Carbon Content</td>
      <td>picogram carbon/cell</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.007702</td>
      <td>0.148502</td>
      <td>mean cellular carbon content based on the equa...</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1187</td>
    </tr>
    <tr>
      <th>15</th>
      <td>prochloro_biomass</td>
      <td>tblSeaFlow</td>
      <td>Prochlorococcus Biomass</td>
      <td>picogram carbon/Î¼L</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.003876</td>
      <td>14.612943</td>
      <td>Qc_mid * Abundance</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1188</td>
    </tr>
    <tr>
      <th>16</th>
      <td>synecho_abundance</td>
      <td>tblSeaFlow</td>
      <td>Synechococcus Abundance</td>
      <td>cells/Î¼L</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.054728</td>
      <td>225.552430</td>
      <td>cell abundance, see https://github.com/armbrus...</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1189</td>
    </tr>
    <tr>
      <th>17</th>
      <td>unknown_diameter</td>
      <td>tblSeaFlow</td>
      <td>Unknown Species Cell Diameter</td>
      <td>micrometer</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.269409</td>
      <td>10.377484</td>
      <td>middle value of mean cell diameter based on Mi...</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1202</td>
    </tr>
    <tr>
      <th>18</th>
      <td>unknown_carbon_content</td>
      <td>tblSeaFlow</td>
      <td>Unknown Species Carbon Content</td>
      <td>picogram carbon/cell</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.005075</td>
      <td>62.589087</td>
      <td>mean cellular carbon content based on the equa...</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1203</td>
    </tr>
    <tr>
      <th>19</th>
      <td>unknown_biomass</td>
      <td>tblSeaFlow</td>
      <td>Unknown Species Biomass</td>
      <td>picogram carbon/Î¼L</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.017999</td>
      <td>1869.242253</td>
      <td>Qc_mid_mean * Abundance</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1204</td>
    </tr>
    <tr>
      <th>20</th>
      <td>total_biomass</td>
      <td>tblSeaFlow</td>
      <td>Total Biomass</td>
      <td>picogram carbon/Î¼L</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Physics</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.000072</td>
      <td>1869.242253</td>
      <td>Qc_mid_mean * Abundance</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1205</td>
    </tr>
    <tr>
      <th>21</th>
      <td>croco_biomass</td>
      <td>tblSeaFlow</td>
      <td>Crocosphaera Biomass</td>
      <td>picogram carbon/Î¼L</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.000072</td>
      <td>296.279373</td>
      <td>Qc_mid_mean * Abundance</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1196</td>
    </tr>
    <tr>
      <th>22</th>
      <td>picoeuk_abundance</td>
      <td>tblSeaFlow</td>
      <td>Picoeukaryote Abundance</td>
      <td>cells/Î¼L</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.017285</td>
      <td>267.632734</td>
      <td>cell abundance, see https://github.com/armbrus...</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1197</td>
    </tr>
    <tr>
      <th>23</th>
      <td>picoeuk_diameter</td>
      <td>tblSeaFlow</td>
      <td>Picoeukaryote Cell Diameter</td>
      <td>micrometer</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.421244</td>
      <td>10.363077</td>
      <td>middle value of mean cell diameter based on Mi...</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1198</td>
    </tr>
    <tr>
      <th>24</th>
      <td>picoeuk_carbon_content</td>
      <td>tblSeaFlow</td>
      <td>Picoeukaryote Carbon Content</td>
      <td>picogram carbon/cell</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.016081</td>
      <td>62.365270</td>
      <td>mean cellular carbon content based on the equa...</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1199</td>
    </tr>
    <tr>
      <th>25</th>
      <td>picoeuk_biomass</td>
      <td>tblSeaFlow</td>
      <td>Picoeukaryote Biomass</td>
      <td>picogram carbon/Î¼L</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.000537</td>
      <td>1180.708795</td>
      <td>Qc_mid_mean * Abundance</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1200</td>
    </tr>
    <tr>
      <th>26</th>
      <td>unknown_abundance</td>
      <td>tblSeaFlow</td>
      <td>Unknown Species Abundance</td>
      <td>cells/Î¼L</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.083200</td>
      <td>823.082261</td>
      <td>cell abundance, see https://github.com/armbrus...</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1201</td>
    </tr>
    <tr>
      <th>27</th>
      <td>synecho_diameter</td>
      <td>tblSeaFlow</td>
      <td>Synechococcus Cell Diameter</td>
      <td>micrometer</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.600849</td>
      <td>1.461741</td>
      <td>middle value of mean cell diameter based on Mi...</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1190</td>
    </tr>
    <tr>
      <th>28</th>
      <td>synecho_carbon_content</td>
      <td>tblSeaFlow</td>
      <td>Synechococcus Carbon Content</td>
      <td>picogram carbon/cell</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.040197</td>
      <td>0.461124</td>
      <td>mean cellular carbon content based on the equa...</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1191</td>
    </tr>
    <tr>
      <th>29</th>
      <td>synecho_biomass</td>
      <td>tblSeaFlow</td>
      <td>Synechococcus Biomass</td>
      <td>picogram carbon/Î¼L</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.004058</td>
      <td>48.321850</td>
      <td>Qc_mid_mean * Abundance</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1192</td>
    </tr>
    <tr>
      <th>30</th>
      <td>croco_abundance</td>
      <td>tblSeaFlow</td>
      <td>Crocosphaera Abundance</td>
      <td>cells/Î¼L</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.005280</td>
      <td>13.707203</td>
      <td>cell abundance, see https://github.com/armbrus...</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1193</td>
    </tr>
    <tr>
      <th>31</th>
      <td>croco_diameter</td>
      <td>tblSeaFlow</td>
      <td>Crocosphaera Cell Diameter</td>
      <td>micrometer</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.319839</td>
      <td>12.710014</td>
      <td>middle value of mean cell diameter based on Mi...</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1194</td>
    </tr>
    <tr>
      <th>32</th>
      <td>croco_carbon_content</td>
      <td>tblSeaFlow</td>
      <td>Crocosphaera Carbon Content</td>
      <td>picogram carbon/cell</td>
      <td>Observation</td>
      <td>Flow Cytometry</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Three Minutes</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.007901</td>
      <td>105.603608</td>
      <td>mean cellular carbon content based on the equa...</td>
      <td>SeaFlow_merged_2019-09-18</td>
      <td>http://doi.org/10.5281/zenodo.3445407</td>
      <td>FranÃ§ois Ribalet, Annette Hynes and Chris Ber...</td>
      <td>The data sets consist of SeaFlow-based cell ab...</td>
      <td>Data provided by: FranÃ§ois Ribalet, Annette H...</td>
      <td>86</td>
      <td>1195</td>
    </tr>
    <tr>
      <th>33</th>
      <td>salinity</td>
      <td>tblCruise_Salinity</td>
      <td>Salinity</td>
      <td>PSU</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Physics</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>0.037000</td>
      <td>1058.000000</td>
      <td>salinity data comes from multiple sources and ...</td>
      <td>Salinity data for cruises in CMAP</td>
      <td>multiple sources - see tblDataset_References</td>
      <td>Multiple Distributors</td>
      <td>tblCruise_Salinity is a aggregation of multipl...</td>
      <td>NaN</td>
      <td>131</td>
      <td>1971</td>
    </tr>
    <tr>
      <th>34</th>
      <td>sra_experiment</td>
      <td>tblKM1513_HOE_legacy_2A_Dyhrman_Omics</td>
      <td>NCBI SRA experiment name</td>
      <td>NaN</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Genomics</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Accecssion numbers to the metagenomic data at ...</td>
      <td>Diel Trichodesmium metatranscriptomes</td>
      <td>Sonya Dyhrman, LDEO/Columbia Univ.</td>
      <td>Sonya Dyhrman - Columbia University</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>122</td>
      <td>1885</td>
    </tr>
    <tr>
      <th>35</th>
      <td>sra_run</td>
      <td>tblKM1513_HOE_legacy_2A_Dyhrman_Omics</td>
      <td>NCBI SRA sequence runs</td>
      <td>NaN</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Genomics</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Accecssion numbers to the metagenomic data at ...</td>
      <td>Diel Trichodesmium metatranscriptomes</td>
      <td>Sonya Dyhrman, LDEO/Columbia Univ.</td>
      <td>Sonya Dyhrman - Columbia University</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>122</td>
      <td>1886</td>
    </tr>
    <tr>
      <th>36</th>
      <td>sra_bioproject</td>
      <td>tblKM1513_HOE_legacy_2A_Dyhrman_Omics</td>
      <td>NCBI SRA Bioproject number</td>
      <td>NaN</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Genomics</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Accecssion numbers to the metagenomic data at ...</td>
      <td>Diel Trichodesmium metatranscriptomes</td>
      <td>Sonya Dyhrman, LDEO/Columbia Univ.</td>
      <td>Sonya Dyhrman - Columbia University</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>122</td>
      <td>1887</td>
    </tr>
    <tr>
      <th>37</th>
      <td>filter_type</td>
      <td>tblKM1513_HOE_legacy_2A_Dyhrman_Omics</td>
      <td>filter_type</td>
      <td>NaN</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Diel Trichodesmium metatranscriptomes</td>
      <td>Sonya Dyhrman, LDEO/Columbia Univ.</td>
      <td>Sonya Dyhrman - Columbia University</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>122</td>
      <td>1888</td>
    </tr>
    <tr>
      <th>38</th>
      <td>filter_max</td>
      <td>tblKM1513_HOE_legacy_2A_Dyhrman_Omics</td>
      <td>Filter Max</td>
      <td>Âµm</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>5.000000</td>
      <td>5.000000</td>
      <td>NaN</td>
      <td>Diel Trichodesmium metatranscriptomes</td>
      <td>Sonya Dyhrman, LDEO/Columbia Univ.</td>
      <td>Sonya Dyhrman - Columbia University</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>122</td>
      <td>1889</td>
    </tr>
    <tr>
      <th>39</th>
      <td>filter_min</td>
      <td>tblKM1513_HOE_legacy_2A_Dyhrman_Omics</td>
      <td>Filter Min</td>
      <td>Âµm</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Biology</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>5.000000</td>
      <td>5.000000</td>
      <td>NaN</td>
      <td>Diel Trichodesmium metatranscriptomes</td>
      <td>Sonya Dyhrman, LDEO/Columbia Univ.</td>
      <td>Sonya Dyhrman - Columbia University</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>122</td>
      <td>1890</td>
    </tr>
    <tr>
      <th>40</th>
      <td>library_kit</td>
      <td>tblKM1513_HOE_legacy_2A_Dyhrman_Omics</td>
      <td>Sequencing Library Kit</td>
      <td>NaN</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Genomics</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Diel Trichodesmium metatranscriptomes</td>
      <td>Sonya Dyhrman, LDEO/Columbia Univ.</td>
      <td>Sonya Dyhrman - Columbia University</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>122</td>
      <td>1891</td>
    </tr>
    <tr>
      <th>41</th>
      <td>sequence_type</td>
      <td>tblKM1513_HOE_legacy_2A_Dyhrman_Omics</td>
      <td>DNA Sequence Type</td>
      <td>NaN</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Genomics</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Diel Trichodesmium metatranscriptomes</td>
      <td>Sonya Dyhrman, LDEO/Columbia Univ.</td>
      <td>Sonya Dyhrman - Columbia University</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>122</td>
      <td>1892</td>
    </tr>
    <tr>
      <th>42</th>
      <td>sequencing_method</td>
      <td>tblKM1513_HOE_legacy_2A_Dyhrman_Omics</td>
      <td>DNA sequencing method</td>
      <td>NaN</td>
      <td>Observation</td>
      <td>In-Situ</td>
      <td>Reprocessed</td>
      <td>Genomics</td>
      <td>Irregular</td>
      <td>Irregular</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Diel Trichodesmium metatranscriptomes</td>
      <td>Sonya Dyhrman, LDEO/Columbia Univ.</td>
      <td>Sonya Dyhrman - Columbia University</td>
      <td>These datasets are from the SCOPE diel cruise ...</td>
      <td>NaN</td>
      <td>122</td>
      <td>1893</td>
    </tr>
  </tbody>
</table>
<p>43 rows × 34 columns</p>
</div></div>
</div>
<p>Returns a subset of data according to the specified space-time constraints (dt1, dt2, lat1, lat2, lon1, lon2, depth1, depth2). The results are ordered by time, lat, lon, and depth (if exists), respectively.</p>
<blockquote>
<div><p><strong>Parameters:</strong> &gt; <strong>table: string</strong> &gt; Table name (each dataset is stored in a table). A full list of table names can be found in <a class="reference external" href="https://cmap.readthedocs.io/en/latest/user_guide/API_ref/pycmap_api/pycmap_catalog.html#getcatalog">catalog</a>. &gt; &gt; <strong>variable: string</strong> &gt; Variable short name which directly corresponds to a field name in the table. A subset of this variable is returned by this method according to the spatio-temporal cut parameters (below). Pass <strong>‘*’</strong> wild card to retrieve
all fields in a table. A full list of variable short names can be found in <a class="reference external" href="https://cmap.readthedocs.io/en/latest/user_guide/API_ref/pycmap_api/pycmap_catalog.html#getcatalog">catalog</a>. &gt; &gt; <strong>dt1: string</strong> &gt; Start date or datetime. This parameter sets the lower bound of the temporal cut. Example values: ‘2016-05-25’ or ‘2017-12-10 17:25:00’ &gt; &gt; <strong>dt2: string</strong> &gt; End date or datetime. This parameter sets the upper bound of the temporal cut. &gt; &gt; <strong>lat1: float</strong> &gt; Start latitude [degree
N]. This parameter sets the lower bound of the meridional cut. Note latitude ranges from -90° to 90°. &gt; &gt; <strong>lat2: float</strong> &gt; End latitude [degree N]. This parameter sets the upper bound of the meridional cut. Note latitude ranges from -90° to 90°. &gt; &gt; <strong>lon1: float</strong> &gt; Start longitude [degree E]. This parameter sets the lower bound of the zonal cut. Note longitue ranges from -180° to 180°. &gt; &gt; <strong>lon2: float</strong> &gt; End longitude [degree E]. This parameter sets the upper bound of the zonal
cut. Note longitue ranges from -180° to 180°. &gt; &gt; <strong>depth1: float</strong> &gt; Start depth [m]. This parameter sets the lower bound of the vertical cut. Note depth is a positive number (it is 0 at surface and grows towards ocean floor). &gt; &gt; <strong>depth2: float</strong> &gt; End depth [m]. This parameter sets the upper bound of the vertical cut. Note depth is a positive number (it is 0 at surface and grows towards ocean floor).</p>
</div></blockquote>
<blockquote>
<div><p><strong>Returns:</strong> &gt; Pandas dataframe.</p>
</div></blockquote>
</div>
<div class="section" id="Example:">
<h2>Example:<a class="headerlink" href="#Example:" title="Permalink to this headline">¶</a></h2>
<p>This example retrieves a subset of in-situ salinity measurements by <a class="reference external" href="https://cmap.readthedocs.io/en/latest/catalog/datasets/Argo.html#argo">Argo floats</a>.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">api</span><span class="o">.</span><span class="n">space_time</span><span class="p">(</span>
              <span class="n">table</span><span class="o">=</span><span class="s1">&#39;tblArgoMerge_REP&#39;</span><span class="p">,</span>
              <span class="n">variable</span><span class="o">=</span><span class="s1">&#39;argo_merge_salinity_adj&#39;</span><span class="p">,</span>
              <span class="n">dt1</span><span class="o">=</span><span class="s1">&#39;2015-05-01&#39;</span><span class="p">,</span>
              <span class="n">dt2</span><span class="o">=</span><span class="s1">&#39;2015-05-30&#39;</span><span class="p">,</span>
              <span class="n">lat1</span><span class="o">=</span><span class="mi">28</span><span class="p">,</span>
              <span class="n">lat2</span><span class="o">=</span><span class="mi">38</span><span class="p">,</span>
              <span class="n">lon1</span><span class="o">=-</span><span class="mi">71</span><span class="p">,</span>
              <span class="n">lon2</span><span class="o">=-</span><span class="mi">50</span><span class="p">,</span>
              <span class="n">depth1</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span>
              <span class="n">depth2</span><span class="o">=</span><span class="mi">100</span>
              <span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>time</th>
      <th>lat</th>
      <th>lon</th>
      <th>depth</th>
      <th>argo_merge_salinity_adj</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>4.300000</td>
      <td>36.713013</td>
    </tr>
    <tr>
      <th>1</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>6.000000</td>
      <td>36.713013</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>7.600000</td>
      <td>36.710995</td>
    </tr>
    <tr>
      <th>3</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>8.000000</td>
      <td>36.711006</td>
    </tr>
    <tr>
      <th>4</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>10.000000</td>
      <td>36.702999</td>
    </tr>
    <tr>
      <th>5</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>11.700000</td>
      <td>36.696999</td>
    </tr>
    <tr>
      <th>6</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>12.000000</td>
      <td>36.697014</td>
    </tr>
    <tr>
      <th>7</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>14.000000</td>
      <td>36.695004</td>
    </tr>
    <tr>
      <th>8</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>16.000000</td>
      <td>36.694008</td>
    </tr>
    <tr>
      <th>9</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>16.700001</td>
      <td>36.694016</td>
    </tr>
    <tr>
      <th>10</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>18.000000</td>
      <td>36.691010</td>
    </tr>
    <tr>
      <th>11</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>20.000000</td>
      <td>36.681019</td>
    </tr>
    <tr>
      <th>12</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>21.100000</td>
      <td>36.665005</td>
    </tr>
    <tr>
      <th>13</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>22.000000</td>
      <td>36.654999</td>
    </tr>
    <tr>
      <th>14</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>24.000000</td>
      <td>36.644009</td>
    </tr>
    <tr>
      <th>15</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>26.000000</td>
      <td>36.631001</td>
    </tr>
    <tr>
      <th>16</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>26.200001</td>
      <td>36.629002</td>
    </tr>
    <tr>
      <th>17</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>28.000000</td>
      <td>36.622997</td>
    </tr>
    <tr>
      <th>18</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>30.000000</td>
      <td>36.619011</td>
    </tr>
    <tr>
      <th>19</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>31.400000</td>
      <td>36.615009</td>
    </tr>
    <tr>
      <th>20</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>32.000000</td>
      <td>36.611008</td>
    </tr>
    <tr>
      <th>21</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>34.000000</td>
      <td>36.606022</td>
    </tr>
    <tr>
      <th>22</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>36.000000</td>
      <td>36.604004</td>
    </tr>
    <tr>
      <th>23</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>36.099998</td>
      <td>36.603996</td>
    </tr>
    <tr>
      <th>24</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>38.000000</td>
      <td>36.603001</td>
    </tr>
    <tr>
      <th>25</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>40.000000</td>
      <td>36.603004</td>
    </tr>
    <tr>
      <th>26</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>41.599998</td>
      <td>36.602001</td>
    </tr>
    <tr>
      <th>27</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>42.000000</td>
      <td>36.603008</td>
    </tr>
    <tr>
      <th>28</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>44.000000</td>
      <td>36.602997</td>
    </tr>
    <tr>
      <th>29</th>
      <td>2015-05-02T00:00:00</td>
      <td>32.348</td>
      <td>-63.450</td>
      <td>46.000000</td>
      <td>36.603012</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>514</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>58.000000</td>
      <td>36.720104</td>
    </tr>
    <tr>
      <th>515</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>60.099998</td>
      <td>36.722099</td>
    </tr>
    <tr>
      <th>516</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>61.900002</td>
      <td>36.726105</td>
    </tr>
    <tr>
      <th>517</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>62.000000</td>
      <td>36.726089</td>
    </tr>
    <tr>
      <th>518</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>64.000000</td>
      <td>36.731098</td>
    </tr>
    <tr>
      <th>519</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>66.000000</td>
      <td>36.730087</td>
    </tr>
    <tr>
      <th>520</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>66.599998</td>
      <td>36.728104</td>
    </tr>
    <tr>
      <th>521</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>68.000000</td>
      <td>36.719101</td>
    </tr>
    <tr>
      <th>522</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>70.000000</td>
      <td>36.712101</td>
    </tr>
    <tr>
      <th>523</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>71.599998</td>
      <td>36.716095</td>
    </tr>
    <tr>
      <th>524</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>72.000000</td>
      <td>36.718090</td>
    </tr>
    <tr>
      <th>525</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>74.000000</td>
      <td>36.712097</td>
    </tr>
    <tr>
      <th>526</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>76.000000</td>
      <td>36.712101</td>
    </tr>
    <tr>
      <th>527</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>76.500000</td>
      <td>36.712101</td>
    </tr>
    <tr>
      <th>528</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>78.000000</td>
      <td>36.714108</td>
    </tr>
    <tr>
      <th>529</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>80.000000</td>
      <td>36.709110</td>
    </tr>
    <tr>
      <th>530</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>81.300003</td>
      <td>36.703102</td>
    </tr>
    <tr>
      <th>531</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>82.000000</td>
      <td>36.701096</td>
    </tr>
    <tr>
      <th>532</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>84.000000</td>
      <td>36.696098</td>
    </tr>
    <tr>
      <th>533</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>86.000000</td>
      <td>36.689102</td>
    </tr>
    <tr>
      <th>534</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>86.699997</td>
      <td>36.687084</td>
    </tr>
    <tr>
      <th>535</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>88.000000</td>
      <td>36.685108</td>
    </tr>
    <tr>
      <th>536</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>90.000000</td>
      <td>36.681091</td>
    </tr>
    <tr>
      <th>537</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>91.099998</td>
      <td>36.676090</td>
    </tr>
    <tr>
      <th>538</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>92.000000</td>
      <td>36.675102</td>
    </tr>
    <tr>
      <th>539</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>94.000000</td>
      <td>36.672115</td>
    </tr>
    <tr>
      <th>540</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>96.000000</td>
      <td>36.670086</td>
    </tr>
    <tr>
      <th>541</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>96.599998</td>
      <td>36.669094</td>
    </tr>
    <tr>
      <th>542</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>98.000000</td>
      <td>36.670105</td>
    </tr>
    <tr>
      <th>543</th>
      <td>2015-05-29T00:00:00</td>
      <td>28.293</td>
      <td>-68.749</td>
      <td>100.000000</td>
      <td>36.672108</td>
    </tr>
  </tbody>
</table>
<p>544 rows × 5 columns</p>
</div></div>
</div>
<p>This method colocalizes a cruise trajectory with the specified target variables. The matching results rely on the tolerance parameters because these parameters set the matching boundaries between the cruise trajectory and target datasets. Please note that the number of matching entries for each target variable might vary depending on the temporal and spatial resolutions of the target variable. In principle, if the cruise trajectory is fully covered by the target variable’s spatio-temporal range,
there should always be matching results if the tolerance parameters are larger than half of their corresponding spatial/temporal resolutions. Please explore the <a class="reference external" href="https://cmap.readthedocs.io/en/latest/user_guide/API_ref/pycmap_api/pycmap_catalog.html#getcatalog">catalog</a> to find appropriate target variables to colocalize with the desired cruise.</p>
<p>This method returns a dataframe containing the cruise trajectory joined with the target variable(s).</p>
<blockquote>
<div><p><strong>Parameters:</strong> &gt; <strong>cruise: string</strong> &gt; The official cruise name. If applicable, you may also use cruise “nickname” (‘Diel’, ‘Gradients_1’ …). A full list of cruise names can be retrieved using <a class="reference external" href="https://cmap.readthedocs.io/en/latest/user_guide/API_ref/pycmap_api/pycmap_catalog.html#getcatalog">cruise</a> method. &gt; &gt; <strong>targetTables: list of string</strong> &gt; Table names of the target datasets to be matched with the cruise trajectory. Notice cruise trajectory can be matched with multiple target
datasets. A full list of table names can be found in <a class="reference external" href="https://cmap.readthedocs.io/en/latest/user_guide/API_ref/pycmap_api/pycmap_catalog.html#getcatalog">catalog</a>. &gt; &gt; <strong>targetVars: list of string</strong> &gt; Variable short names to be matched with the cruise trajectory. A full list of variable short names can be found in <a class="reference external" href="https://cmap.readthedocs.io/en/latest/user_guide/API_ref/pycmap_api/pycmap_catalog.html#getcatalog">catalog</a>. &gt; &gt; <strong>depth1: float</strong> &gt; Start depth [m]. This parameter sets
the lower bound of the depth cut on the traget datasets. ‘depth1’ and ‘depth2’ allow matching a cruise trajectory (which is at the surface, hopefully!) with traget varaiables at lower depth. Note depth is a positive number (depth is 0 at surface and grows towards ocean floor). &gt; &gt; <strong>depth2: float</strong> &gt; End depth [m]. This parameter sets the upper bound of the depth cut on the traget datasets. Note depth is a positive number (depth is 0 at surface and grows towards ocean floor). &gt; &gt;
<strong>temporalTolerance: list of int</strong> &gt; Temporal tolerance values between the cruise trajectory and target datasets. The size and order of values in this list should match those of targetTables. If only a single integer value is given, that would be applied to all target datasets. This parameter is in day units except when the target variable represents monthly climatology data in which case it is in month units. Notice fractional values are not supported in the current version. &gt; &gt;
<strong>latTolerance: list of float or int</strong> &gt; Spatial tolerance values in meridional direction [deg] between the cruise trajectory and target datasets. The size and order of values in this list should match those of targetTables. If only a single float value is given, that would be applied to all target datasets. A “safe” value for this parameter can be slightly larger than the half of the traget variable’s spatial resolution. &gt; &gt; <strong>lonTolerance: list of float or int</strong> &gt; Spatial tolerance
values in zonal direction [deg] between the cruise trajectory and target datasets. The size and order of values in this list should match those of targetTables. If only a single float value is given, that would be applied to all target datasets. A “safe” value for this parameter can be slightly larger than the half of the traget variable’s spatial resolution. &gt; &gt; <strong>depthTolerance: list of float or int</strong> &gt; Spatial tolerance values in vertical direction [m] between the cruise trajectory and
target datasets. The size and order of values in this list should match those of targetTables. If only a single float value is given, that would be applied to all target datasets.</p>
</div></blockquote>
<blockquote>
<div><p><strong>Returns:</strong> &gt; Pandas dataframe.</p>
</div></blockquote>
</div>
<div class="section" id="Example:">
<h2>Example:<a class="headerlink" href="#Example:" title="Permalink to this headline">¶</a></h2>
<p>Colocalizes the Gradients_1 cruise with prochloro_abundance and prokaryote_c01_darwin_clim variables from the Seaflow and Darwin (climatology) Data sets, respectively.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="o">%</span><span class="k">matplotlib</span> inline
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">import</span> <span class="nn">pycmap</span>

<span class="n">api</span> <span class="o">=</span> <span class="n">pycmap</span><span class="o">.</span><span class="n">API</span><span class="p">()</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">api</span><span class="o">.</span><span class="n">along_track</span><span class="p">(</span>
                    <span class="n">cruise</span><span class="o">=</span><span class="s1">&#39;gradients_3&#39;</span><span class="p">,</span>
                    <span class="n">targetTables</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;tblSeaFlow&#39;</span><span class="p">,</span> <span class="s1">&#39;tblDarwin_Nutrient_Climatology&#39;</span><span class="p">],</span>
                    <span class="n">targetVars</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;prochloro_abundance&#39;</span><span class="p">,</span> <span class="s1">&#39;PO4_darwin_clim&#39;</span><span class="p">],</span>
                    <span class="n">depth1</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span>
                    <span class="n">depth2</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span>
                    <span class="n">temporalTolerance</span><span class="o">=</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="mi">0</span><span class="p">],</span>
                    <span class="n">latTolerance</span><span class="o">=</span><span class="p">[</span><span class="mf">0.01</span><span class="p">,</span> <span class="mf">0.25</span><span class="p">],</span>
                    <span class="n">lonTolerance</span><span class="o">=</span><span class="p">[</span><span class="mf">0.01</span><span class="p">,</span> <span class="mf">0.25</span><span class="p">],</span>
                    <span class="n">depthTolerance</span><span class="o">=</span><span class="p">[</span><span class="mi">5</span><span class="p">,</span> <span class="mi">5</span><span class="p">]</span>
                    <span class="p">)</span>




<span class="c1">################# Simple Plot #################</span>
<span class="n">fig</span><span class="p">,</span> <span class="n">ax1</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">()</span>
<span class="n">ax2</span> <span class="o">=</span> <span class="n">ax1</span><span class="o">.</span><span class="n">twinx</span><span class="p">()</span>
<span class="n">c1</span><span class="p">,</span> <span class="n">c2</span> <span class="o">=</span> <span class="s1">&#39;firebrick&#39;</span><span class="p">,</span> <span class="s1">&#39;slateblue&#39;</span>
<span class="n">t1</span><span class="p">,</span> <span class="n">t2</span> <span class="o">=</span> <span class="s1">&#39;tblSeaFlow&#39;</span><span class="p">,</span> <span class="s1">&#39;tblDarwin_Nutrient_Climatology&#39;</span>
<span class="n">v1</span><span class="p">,</span> <span class="n">v2</span> <span class="o">=</span> <span class="s1">&#39;prochloro_abundance&#39;</span><span class="p">,</span> <span class="s1">&#39;PO4_darwin_clim&#39;</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;lat&#39;</span><span class="p">],</span> <span class="n">df</span><span class="p">[</span><span class="n">v1</span><span class="p">],</span> <span class="s1">&#39;o&#39;</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">c1</span><span class="p">,</span> <span class="n">markeredgewidth</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s1">&#39;SeaFlow&#39;</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mf">0.2</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span> <span class="n">labelcolor</span><span class="o">=</span><span class="s1">&#39;r&#39;</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="n">v1</span> <span class="o">+</span> <span class="n">api</span><span class="o">.</span><span class="n">get_unit</span><span class="p">(</span><span class="n">t1</span><span class="p">,</span> <span class="n">v1</span><span class="p">),</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;r&#39;</span><span class="p">)</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;lat&#39;</span><span class="p">],</span> <span class="n">df</span><span class="p">[</span><span class="n">v2</span><span class="p">],</span> <span class="s1">&#39;o&#39;</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">c2</span><span class="p">,</span> <span class="n">markeredgewidth</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s1">&#39;Darwin&#39;</span><span class="p">,</span> <span class="n">alpha</span><span class="o">=</span><span class="mf">0.2</span><span class="p">)</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">tick_params</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="s1">&#39;y&#39;</span><span class="p">,</span> <span class="n">labelcolor</span><span class="o">=</span><span class="s1">&#39;b&#39;</span><span class="p">)</span>
<span class="n">ax2</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="n">v2</span> <span class="o">+</span> <span class="n">api</span><span class="o">.</span><span class="n">get_unit</span><span class="p">(</span><span class="n">t2</span><span class="p">,</span> <span class="n">v2</span><span class="p">),</span> <span class="n">color</span><span class="o">=</span><span class="s1">&#39;b&#39;</span><span class="p">)</span>
<span class="n">ax1</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s1">&#39;Latitude&#39;</span><span class="p">)</span>
<span class="n">fig</span><span class="o">.</span><span class="n">tight_layout</span><span class="p">()</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<script type="application/vnd.jupyter.widget-view+json">{"model_id": "34ee738b42d541dfb89ff38d2f504228", "version_major": 2, "version_minor": 0}</script></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>

1: prochloro_abundance matched.

2: PO4_darwin_clim matched.

</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<blockquote>
<div><dl class="field-list simple">
<dt class="field-odd">fancy</dt>
<dd class="field-odd"><p></p></dd>
</dl>
<img alt="../../../../_images/user_guide_API_ref_pycmap_api_presentations_SCOPE2019_28_2.png" src="../../../../_images/user_guide_API_ref_pycmap_api_presentations_SCOPE2019_28_2.png" />
</div></blockquote>
<p>Table of Contents ## <a class="reference external" href="https://cmap.readthedocs.io/en/latest/user_guide/API_ref/pycmap_api/pycmap_query.html#query">query(query)</a> Simons CMAP datasets are hosted in a SQL database and pycmap package provides the user with a number of pre-developed methods to extract and retrieve subsets of the data. The rest of this documentation is dedicated to explore and explain these methods. In addition to the pre-developed methods, we intend to leave the database open to custom scan queries for</p>
</div>
</div>
<p>interested users. This method takes a custom SQL query statement and returns the results in form of a Pandas dataframe. The full list of table names and variable names (fields) can be obtained using the <a class="reference external" href="Catalog.ipynb">get_catalog()</a> method. In fact, one may use this very method to retrieve the table and field names: <code class="docutils literal notranslate"><span class="pre">query('EXEC</span> <span class="pre">uspCatalog')</span></code>. A Dataset is stored in a table and each table field represents a variable. All data tables have the following fields:</p>
<ul class="simple">
<li><p>[time] [date or datetime] NOT NULL,</p></li>
<li><p>[lat] [float] NOT NULL,</p></li>
<li><p>[lon] [float] NOT NULL,</p></li>
<li><p>[depth] [float] NOT NULL,</p></li>
</ul>
</div>
<div class="section" id="Note:">
<h2>Note:<a class="headerlink" href="#Note:" title="Permalink to this headline">¶</a></h2>
<p>Tables which represent a climatological dataset, such as ‘tblDarwin_Nutrient_Climatology’, will not have a ‘time’ field. Also, if a table represents a surface dataset, such as satellite products, there would be no ‘depth’ field. ‘depth’ is a positive number in meters unit; it is zero at the surface growing towards the ocean’s floor. ‘lat’ and ‘lon’ are in degrees units, ranging from -90° to 90° and -180° to 180°, respectively.</p>
<p>Please keep in mind that some of the datasets are massive in size (10s of TB), avoid queries without WHERE clause (<code class="docutils literal notranslate"><span class="pre">SELECT</span> <span class="pre">*</span> <span class="pre">FROM</span> <span class="pre">TABLENAME</span></code>). Always try to add some constraints on time, lat, lon, and depth fields (see the basic examples below).</p>
<p>Moreover, the database hosts a wide range of predefined stored procedures and functions to streamline nearly all CMAP data services. For instance retrieving the catalog information is achieved using a single call of this procedure: <em>uspCatalog</em>. These predefined procedures can be called using the pycmap package (see example below). Alternatively, one may use any SQL client to execute these procedures to retrieve and visualize data (examples: <a class="reference external" href="https://docs.microsoft.com/en-us/sql/azure-data-studio/download?view=sql-server-ver15">Azure Data
Studio</a>, or <a class="reference external" href="https://plot.ly/free-sql-client-download/">Plotly Falcon</a>). Using the predefined procedures all CMAP data services are centralized at the database layer which dramatically facilitates the process of developing apps with different programming languages (pycmap, web app, cmap4r, …). Please note that you can improve the current procedures or add new procedures by contributing at the <a class="reference external" href="https://github.com/simonscmap/DB">CMAP
database repository</a>.  Below is a selected list of stored procedures and functions, their arguments will be described in more details subsequently:</p>
<ul class="simple">
<li><p>uspCatalog</p></li>
<li><p>uspSpaceTime</p></li>
<li><p>uspTimeSeries</p></li>
<li><p>uspDepthProfile</p></li>
<li><p>uspSectionMap</p></li>
<li><p>uspCruises</p></li>
<li><p>uspCruiseByName</p></li>
<li><p>uspCruiseBounds</p></li>
<li><p>uspWeekly</p></li>
<li><p>uspMonthly</p></li>
<li><p>uspQuarterly</p></li>
<li><p>uspAnnual</p></li>
<li><p>uspMatch</p></li>
<li><p>udfDatasetReferences</p></li>
<li><p>udfMetaData_NoRef</p></li>
</ul>
<p>Happy SQL Injection!</p>
</div>
<div class="section" id="Example:">
<h2>Example:<a class="headerlink" href="#Example:" title="Permalink to this headline">¶</a></h2>
<p>A sample stored procedure returning the list of all cruises hosted by Simons CMAP.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">api</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s1">&#39;EXEC uspCruises&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>ID</th>
      <th>Nickname</th>
      <th>Name</th>
      <th>Ship_Name</th>
      <th>Start_Time</th>
      <th>End_Time</th>
      <th>Lat_Min</th>
      <th>Lat_Max</th>
      <th>Lon_Min</th>
      <th>Lon_Max</th>
      <th>Chief_Name</th>
      <th>Keywords</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>616</td>
      <td>AMT01</td>
      <td>AMT01</td>
      <td>RRS James Clark Ross</td>
      <td>1995-09-20T17:51:00.000Z</td>
      <td>1995-10-25T10:06:00.000Z</td>
      <td>-51.942200</td>
      <td>53.588600</td>
      <td>-58.501600</td>
      <td>1.932000</td>
      <td>David Robins</td>
      <td>(East), (SATG), (STGE), (TRPG), (WWDR), 1995, ...</td>
    </tr>
    <tr>
      <th>1</th>
      <td>617</td>
      <td>AMT02</td>
      <td>AMT02</td>
      <td>RRS James Clark Ross</td>
      <td>1996-04-22T12:10:00.000Z</td>
      <td>1996-05-22T07:16:00.000Z</td>
      <td>-51.691900</td>
      <td>50.292900</td>
      <td>-57.830400</td>
      <td>-4.202700</td>
      <td>David Robins</td>
      <td>(East), (SATG), (STGE), (TRPG), (WWDR), 1996, ...</td>
    </tr>
    <tr>
      <th>2</th>
      <td>618</td>
      <td>AMT03</td>
      <td>AMT03</td>
      <td>RRS James Clark Ross</td>
      <td>1996-09-21T12:18:00.000Z</td>
      <td>1996-10-25T13:20:00.000Z</td>
      <td>-51.971500</td>
      <td>50.801100</td>
      <td>-58.480500</td>
      <td>-0.971600</td>
      <td>Tony Bale</td>
      <td>(East), (SATG), (STGE), (TRPG), (WWDR), 1996, ...</td>
    </tr>
    <tr>
      <th>3</th>
      <td>619</td>
      <td>AMT04</td>
      <td>AMT04</td>
      <td>RRS James Clark Ross</td>
      <td>1997-04-21T12:07:00.000Z</td>
      <td>1997-05-27T00:05:00.000Z</td>
      <td>-51.692100</td>
      <td>53.405000</td>
      <td>-57.836200</td>
      <td>2.001200</td>
      <td>Tony Bale</td>
      <td>(East), (SATG), (STGE), (TRPG), (WWDR), 1997, ...</td>
    </tr>
    <tr>
      <th>4</th>
      <td>620</td>
      <td>AMT05</td>
      <td>AMT05</td>
      <td>RRS James Clark Ross</td>
      <td>1997-09-14T04:00:00.000Z</td>
      <td>1997-10-17T13:13:00.000Z</td>
      <td>-51.700100</td>
      <td>53.550300</td>
      <td>-58.291100</td>
      <td>1.938300</td>
      <td>Jim Aiken</td>
      <td>(East), (SATG), (STGE), (TRPG), (WWDR), 1997, ...</td>
    </tr>
    <tr>
      <th>5</th>
      <td>621</td>
      <td>AMT06</td>
      <td>AMT06</td>
      <td>RRS James Clark Ross</td>
      <td>1998-05-15T12:01:00.000Z</td>
      <td>1998-06-16T08:41:00.000Z</td>
      <td>-33.864100</td>
      <td>53.586400</td>
      <td>-20.013100</td>
      <td>18.354300</td>
      <td>Jim Aiken</td>
      <td>(EACB), (East), (SATG), (STGE), (TRPG), (WWDR)...</td>
    </tr>
    <tr>
      <th>6</th>
      <td>622</td>
      <td>AMT07</td>
      <td>AMT07</td>
      <td>RRS James Clark Ross</td>
      <td>1998-09-13T16:00:00.000Z</td>
      <td>1998-10-25T19:55:00.000Z</td>
      <td>-51.691900</td>
      <td>50.808100</td>
      <td>-57.829700</td>
      <td>-0.973000</td>
      <td>Jim Aiken</td>
      <td>(EACB), (East), (SATG), (STGE), (TRPG), (WWDR)...</td>
    </tr>
    <tr>
      <th>7</th>
      <td>623</td>
      <td>AMT08</td>
      <td>AMT08</td>
      <td>RRS James Clark Ross</td>
      <td>1999-04-23T18:51:01.000Z</td>
      <td>1999-06-07T08:02:01.000Z</td>
      <td>-51.692300</td>
      <td>53.503200</td>
      <td>-57.829900</td>
      <td>1.914800</td>
      <td>Nigel Rees</td>
      <td>(East), (SATG), (STGE), (TRPG), (WWDR), 1999, ...</td>
    </tr>
    <tr>
      <th>8</th>
      <td>624</td>
      <td>AMT09</td>
      <td>AMT09</td>
      <td>RRS James Clark Ross</td>
      <td>1999-09-15T08:07:30.000Z</td>
      <td>1999-10-12T14:45:30.000Z</td>
      <td>-36.006800</td>
      <td>53.587000</td>
      <td>-51.412700</td>
      <td>1.950500</td>
      <td>Nigel Rees</td>
      <td>(East), (SATG), (STGE), (TRPG), (WWDR), 1999, ...</td>
    </tr>
    <tr>
      <th>9</th>
      <td>625</td>
      <td>AMT10</td>
      <td>AMT10</td>
      <td>RRS James Clark Ross</td>
      <td>2000-04-11T12:01:00.000Z</td>
      <td>2000-05-07T14:57:00.000Z</td>
      <td>-35.113900</td>
      <td>51.839700</td>
      <td>-56.224700</td>
      <td>1.826700</td>
      <td>Chris Gallienne</td>
      <td>(East), (SATG), (STGE), (TRPG), (WWDR), 2000, ...</td>
    </tr>
    <tr>
      <th>10</th>
      <td>626</td>
      <td>AMT11</td>
      <td>AMT11</td>
      <td>RRS James Clark Ross</td>
      <td>2000-09-13T18:10:10.000Z</td>
      <td>2000-10-11T10:50:10.000Z</td>
      <td>-35.097600</td>
      <td>50.793100</td>
      <td>-56.179500</td>
      <td>-0.962100</td>
      <td>Chris Gallienne</td>
      <td>(East), (SATG), (STGE), (TRPG), (WWDR), 2000, ...</td>
    </tr>
    <tr>
      <th>11</th>
      <td>627</td>
      <td>AMT12</td>
      <td>AMT12</td>
      <td>RRS James Clark Ross</td>
      <td>2003-05-13T12:24:00.000Z</td>
      <td>2003-06-15T10:54:00.000Z</td>
      <td>-50.317730</td>
      <td>49.426400</td>
      <td>-53.682210</td>
      <td>-4.820580</td>
      <td>NaN</td>
      <td>(East), (SATG), (STGE), (TRPG), (WWDR), 2003, ...</td>
    </tr>
    <tr>
      <th>12</th>
      <td>628</td>
      <td>AMT13</td>
      <td>AMT13</td>
      <td>RRS James Clark Ross</td>
      <td>2003-09-11T07:07:00.000Z</td>
      <td>2003-10-13T18:09:00.000Z</td>
      <td>-51.690430</td>
      <td>51.107920</td>
      <td>-57.829850</td>
      <td>1.535970</td>
      <td>Carol Robinson</td>
      <td>(EACB), (East), (SATG), (STGE), (TRPG), (WWDR)...</td>
    </tr>
    <tr>
      <th>13</th>
      <td>629</td>
      <td>AMT14</td>
      <td>AMT14</td>
      <td>RRS James Clark Ross</td>
      <td>2004-04-28T19:12:00.000Z</td>
      <td>2004-05-30T15:11:00.000Z</td>
      <td>-51.691670</td>
      <td>49.382650</td>
      <td>-57.829300</td>
      <td>-5.766520</td>
      <td>Patrick Holligan</td>
      <td>(East), (SATG), (STGE), (STGW), (TRPG), (West)...</td>
    </tr>
    <tr>
      <th>14</th>
      <td>630</td>
      <td>AMT15</td>
      <td>AMT15</td>
      <td>RRS James Clark Ross</td>
      <td>2004-09-17T13:08:40.000Z</td>
      <td>2004-10-27T17:51:40.000Z</td>
      <td>-40.016900</td>
      <td>50.748200</td>
      <td>-25.015900</td>
      <td>16.464600</td>
      <td>Andy Rees</td>
      <td>(EACB), (East), (SATG), (STGE), (TRPG), (WWDR)...</td>
    </tr>
    <tr>
      <th>15</th>
      <td>631</td>
      <td>AMT16</td>
      <td>AMT16</td>
      <td>RRS James Clark Ross</td>
      <td>2005-05-20T12:00:00.000Z</td>
      <td>2005-06-28T13:00:00.000Z</td>
      <td>-33.888700</td>
      <td>48.529500</td>
      <td>-46.080600</td>
      <td>18.458300</td>
      <td>Tony Bale</td>
      <td>(East), (SATG), (STGE), (STGW), (TRPG), (West)...</td>
    </tr>
    <tr>
      <th>16</th>
      <td>632</td>
      <td>AMT17</td>
      <td>AMT17</td>
      <td>RRS James Clark Ross</td>
      <td>2005-10-15T08:25:00.000Z</td>
      <td>2005-11-26T05:50:00.000Z</td>
      <td>-35.866600</td>
      <td>55.969500</td>
      <td>-39.097500</td>
      <td>18.871200</td>
      <td>Patrick Holligan</td>
      <td>(East), (SATG), (STGE), (STGW), (TRPG), (West)...</td>
    </tr>
    <tr>
      <th>17</th>
      <td>633</td>
      <td>AMT19</td>
      <td>AMT19</td>
      <td>RRS James Clark Ross</td>
      <td>2009-10-13T12:12:00.000Z</td>
      <td>2009-11-30T11:06:00.000Z</td>
      <td>-51.590100</td>
      <td>50.157800</td>
      <td>-65.700600</td>
      <td>-4.998000</td>
      <td>Andy Rees</td>
      <td>(East), (SATG), (STGE), (STGW), (TRPG), (West)...</td>
    </tr>
    <tr>
      <th>18</th>
      <td>634</td>
      <td>AMT20</td>
      <td>AMT20</td>
      <td>RRS James Clark Ross</td>
      <td>2010-10-12T10:40:00.000Z</td>
      <td>2010-11-22T21:10:00.000Z</td>
      <td>-48.107500</td>
      <td>50.754300</td>
      <td>-56.124600</td>
      <td>-1.361000</td>
      <td>Andy Rees</td>
      <td>(East), (SATG), (STGE), (STGW), (TRPG), (West)...</td>
    </tr>
    <tr>
      <th>19</th>
      <td>635</td>
      <td>AMT21</td>
      <td>AMT21</td>
      <td>RRS James Clark Ross</td>
      <td>2011-09-28T08:07:00.000Z</td>
      <td>2011-11-11T17:59:00.000Z</td>
      <td>-52.054900</td>
      <td>51.507000</td>
      <td>-67.102000</td>
      <td>-2.704000</td>
      <td>Glen Tarran</td>
      <td>(East), (SATG), (STGE), (STGW), (TRPG), (West)...</td>
    </tr>
    <tr>
      <th>20</th>
      <td>636</td>
      <td>AMT22</td>
      <td>AMT22</td>
      <td>RRS James Clark Ross</td>
      <td>2012-10-10T08:00:00.000Z</td>
      <td>2012-11-20T20:01:00.000Z</td>
      <td>-47.149800</td>
      <td>50.891800</td>
      <td>-54.826400</td>
      <td>-1.291700</td>
      <td>Glen Tarran</td>
      <td>(East), (SATG), (STGE), (STGW), (TRPG), (West)...</td>
    </tr>
    <tr>
      <th>21</th>
      <td>564</td>
      <td>MBARI_1</td>
      <td>CN11ID</td>
      <td>R/V Western Flyer</td>
      <td>2011-09-30T17:55:09.000Z</td>
      <td>2011-10-11T17:10:06.000Z</td>
      <td>33.648765</td>
      <td>36.797673</td>
      <td>-128.049333</td>
      <td>-121.846333</td>
      <td>Alexandra Worden</td>
      <td>2011, Alexandra, Aquarium, Bay, California, CC...</td>
    </tr>
    <tr>
      <th>22</th>
      <td>570</td>
      <td>MBARI_2</td>
      <td>CN12ID</td>
      <td>R/V Western Flyer</td>
      <td>2012-09-11T03:13:48.000Z</td>
      <td>2012-09-20T15:02:06.000Z</td>
      <td>35.458867</td>
      <td>36.800083</td>
      <td>-124.907183</td>
      <td>-121.805500</td>
      <td>Alexandra Worden</td>
      <td>2012, Alexandra, Aquarium, Bay, California, CC...</td>
    </tr>
    <tr>
      <th>23</th>
      <td>575</td>
      <td>MBARI_3</td>
      <td>CN13ID</td>
      <td>R/V Western Flyer</td>
      <td>2013-10-08T18:04:58.000Z</td>
      <td>2013-10-18T14:09:55.000Z</td>
      <td>33.156300</td>
      <td>36.795400</td>
      <td>-129.429100</td>
      <td>-121.853000</td>
      <td>Alexandra Worden</td>
      <td>2013, Alexandra, Aquarium, Bay, California, CC...</td>
    </tr>
    <tr>
      <th>24</th>
      <td>601</td>
      <td>SCOPE_Falkor1</td>
      <td>FK180310-1</td>
      <td>R/V Falkor</td>
      <td>2018-03-11T19:43:14.000Z</td>
      <td>2018-03-23T06:18:32.000Z</td>
      <td>20.980100</td>
      <td>22.720700</td>
      <td>-158.352000</td>
      <td>-155.085000</td>
      <td>Samuel Wilson</td>
      <td>2018, and, Aquarium, Armbrust, Autonomous, Bas...</td>
    </tr>
    <tr>
      <th>25</th>
      <td>602</td>
      <td>SCOPE_Falkor2</td>
      <td>FK180310-2</td>
      <td>R/V Falkor</td>
      <td>2018-03-28T02:26:35.000Z</td>
      <td>2018-04-10T02:18:12.000Z</td>
      <td>21.167400</td>
      <td>24.710400</td>
      <td>-160.981000</td>
      <td>-156.268600</td>
      <td>Samuel Wilson</td>
      <td>2018, and, Apr, April, Aquarium, Armbrust, Aut...</td>
    </tr>
    <tr>
      <th>26</th>
      <td>574</td>
      <td>KiloMoana_1</td>
      <td>KM1314</td>
      <td>R/V Kilo Moana</td>
      <td>2013-08-08T00:00:21.000Z</td>
      <td>2013-09-05T14:53:15.000Z</td>
      <td>21.229300</td>
      <td>50.044100</td>
      <td>-158.527000</td>
      <td>-124.253000</td>
      <td>Anitra Ingalls</td>
      <td>(East), 2013, Anitra, Aug, August, California,...</td>
    </tr>
    <tr>
      <th>27</th>
      <td>576</td>
      <td>SCOPE_1</td>
      <td>KM1427</td>
      <td>R/V Kilo Moana</td>
      <td>2014-12-08T22:53:34.000Z</td>
      <td>2014-12-12T02:33:13.000Z</td>
      <td>21.222900</td>
      <td>22.872300</td>
      <td>-158.334500</td>
      <td>-157.865500</td>
      <td>Samuel Wilson</td>
      <td>2014, Armbrust, chroco, CMAP, CMORE, C-MORE, c...</td>
    </tr>
    <tr>
      <th>28</th>
      <td>577</td>
      <td>SCOPE_2</td>
      <td>KM1502</td>
      <td>R/V Kilo Moana</td>
      <td>2015-03-21T04:52:20.000Z</td>
      <td>2015-03-29T05:21:56.000Z</td>
      <td>21.981300</td>
      <td>46.130700</td>
      <td>-156.692100</td>
      <td>-124.302300</td>
      <td>NaN</td>
      <td>2015, and, Armbrust, California, CCAL, chroco,...</td>
    </tr>
    <tr>
      <th>29</th>
      <td>578</td>
      <td>SCOPE_3</td>
      <td>KM1508</td>
      <td>R/V Kilo Moana</td>
      <td>2015-05-22T22:53:01.000Z</td>
      <td>2015-05-26T16:43:23.000Z</td>
      <td>21.244800</td>
      <td>22.868800</td>
      <td>-158.369800</td>
      <td>-157.887600</td>
      <td>Daniel Sadler</td>
      <td>2015, Aloha, and, Armbrust, chroco, CMAP, Coll...</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>53</th>
      <td>588</td>
      <td>SCOPE_15</td>
      <td>KOK1604</td>
      <td>R/V Kaimikai O Kanaloa</td>
      <td>2016-04-13T14:13:48.000Z</td>
      <td>2016-04-17T01:36:58.000Z</td>
      <td>21.571500</td>
      <td>22.771500</td>
      <td>-158.398100</td>
      <td>-157.901400</td>
      <td>Margo Edwards</td>
      <td>2016, Apr, April, Armbrust, chroco, CMAP, croc...</td>
    </tr>
    <tr>
      <th>54</th>
      <td>589</td>
      <td>Gradients_1</td>
      <td>KOK1606</td>
      <td>R/V Kaimikai O Kanaloa</td>
      <td>2016-04-20T00:04:37.000Z</td>
      <td>2016-05-04T02:33:45.000Z</td>
      <td>21.454200</td>
      <td>37.886400</td>
      <td>-158.335500</td>
      <td>-157.858000</td>
      <td>Virginia Armbrust</td>
      <td>(via, 1, 13C, 13C_fixation_rate_mean, 13C_fixa...</td>
    </tr>
    <tr>
      <th>55</th>
      <td>590</td>
      <td>SCOPE_17</td>
      <td>KOK1607</td>
      <td>R/V Kaimikai O Kanaloa</td>
      <td>2016-05-12T16:58:49.000Z</td>
      <td>2016-05-14T06:50:34.000Z</td>
      <td>22.809200</td>
      <td>24.487300</td>
      <td>-157.385200</td>
      <td>-155.921400</td>
      <td>Benedetto Barone</td>
      <td>&gt;5, 2016, ÃÂµm, and, Armbrust, Barone, Benede...</td>
    </tr>
    <tr>
      <th>56</th>
      <td>591</td>
      <td>SCOPE_18</td>
      <td>KOK1608</td>
      <td>R/V Kaimikai O Kanaloa</td>
      <td>2016-07-10T20:25:23.000Z</td>
      <td>2016-07-14T09:10:17.000Z</td>
      <td>21.251100</td>
      <td>22.774800</td>
      <td>-158.365800</td>
      <td>-157.908200</td>
      <td>Matthew Church</td>
      <td>2016, Armbrust, chroco, Church, CMAP, croco_ab...</td>
    </tr>
    <tr>
      <th>57</th>
      <td>592</td>
      <td>SCOPE_19</td>
      <td>KOK1609</td>
      <td>R/V Kaimikai O Kanaloa</td>
      <td>2016-08-08T19:33:41.000Z</td>
      <td>2016-08-12T08:31:17.000Z</td>
      <td>21.715600</td>
      <td>22.804300</td>
      <td>-158.363400</td>
      <td>-157.884200</td>
      <td>Terry Kerby</td>
      <td>2016, Armbrust, Aug, August, chroco, CMAP, cro...</td>
    </tr>
    <tr>
      <th>58</th>
      <td>603</td>
      <td>HOT301</td>
      <td>KOK1801</td>
      <td>R/V Kaimikai O Kanaloa</td>
      <td>2018-04-16T20:54:14.000Z</td>
      <td>2018-04-20T08:15:41.000Z</td>
      <td>21.324600</td>
      <td>22.879900</td>
      <td>-158.359900</td>
      <td>-157.885300</td>
      <td>David Karl</td>
      <td>2018, Apr, April, CMAP, cruises, data, David, ...</td>
    </tr>
    <tr>
      <th>59</th>
      <td>604</td>
      <td>HOT302</td>
      <td>KOK1803</td>
      <td>R/V Kaimikai O Kanaloa</td>
      <td>2018-05-14T22:09:23.000Z</td>
      <td>2018-05-18T06:24:32.000Z</td>
      <td>21.343800</td>
      <td>22.861100</td>
      <td>-158.339000</td>
      <td>-157.889000</td>
      <td>David Karl</td>
      <td>2018, CMAP, cruises, data, David, for, Gyre, H...</td>
    </tr>
    <tr>
      <th>60</th>
      <td>605</td>
      <td>HOT303</td>
      <td>KOK1804</td>
      <td>R/V Kaimikai O Kanaloa</td>
      <td>2018-06-25T19:57:47.000Z</td>
      <td>2018-06-29T10:28:24.000Z</td>
      <td>21.312900</td>
      <td>22.837900</td>
      <td>-158.370366</td>
      <td>-157.898300</td>
      <td>David Karl</td>
      <td>2018, CMAP, cruises, data, David, for, Gyre, H...</td>
    </tr>
    <tr>
      <th>61</th>
      <td>606</td>
      <td>KOK1805</td>
      <td>KOK1805</td>
      <td>R/V Kaimikai O Kanaloa</td>
      <td>2018-07-07T23:30:25.000Z</td>
      <td>2018-07-13T23:57:47.000Z</td>
      <td>18.847300</td>
      <td>21.024800</td>
      <td>-157.685600</td>
      <td>-154.605900</td>
      <td>Jacqueline Caplan-Auerbach</td>
      <td>2018, Caplan-Auerbach, CMAP, cruises, data, fo...</td>
    </tr>
    <tr>
      <th>62</th>
      <td>607</td>
      <td>KOK1806</td>
      <td>KOK1806</td>
      <td>R/V Kaimikai O Kanaloa</td>
      <td>2018-07-13T22:24:40.000Z</td>
      <td>2018-07-17T07:48:39.000Z</td>
      <td>18.689500</td>
      <td>20.739700</td>
      <td>-157.350100</td>
      <td>-154.361700</td>
      <td>David Karl</td>
      <td>&gt;5, 2018, Aftermath,, Alkalinity, Aquatic, Arm...</td>
    </tr>
    <tr>
      <th>63</th>
      <td>593</td>
      <td>Gradients_2</td>
      <td>MGL1704</td>
      <td>R/V Langseth</td>
      <td>2017-05-26T22:34:53.000Z</td>
      <td>2017-06-13T18:07:25.000Z</td>
      <td>21.300800</td>
      <td>42.427000</td>
      <td>-158.347500</td>
      <td>-157.630100</td>
      <td>Virginia Armbrust</td>
      <td>(via, 13C, 13C_fixation_rate_mean, 13C_fixatio...</td>
    </tr>
    <tr>
      <th>64</th>
      <td>610</td>
      <td>GP15_1</td>
      <td>RR1814</td>
      <td>R/V Roger Revelle</td>
      <td>2018-09-19T15:50:13.000Z</td>
      <td>2018-10-21T15:58:18.000Z</td>
      <td>19.680600</td>
      <td>56.059600</td>
      <td>-156.974100</td>
      <td>-127.627000</td>
      <td>Gregory Cutter</td>
      <td>(East), 2018, Alaska, ALSK, California, CCAL, ...</td>
    </tr>
    <tr>
      <th>65</th>
      <td>611</td>
      <td>GP15_2</td>
      <td>RR1815</td>
      <td>R/V Roger Revelle</td>
      <td>2018-10-23T20:34:14.000Z</td>
      <td>2018-11-24T06:12:31.000Z</td>
      <td>-20.001200</td>
      <td>19.730300</td>
      <td>-155.258000</td>
      <td>-150.621500</td>
      <td>Gregory Cutter</td>
      <td>2018, CMAP, Countercurrent, cruises, Cutter, d...</td>
    </tr>
    <tr>
      <th>66</th>
      <td>554</td>
      <td>Thompson_0</td>
      <td>TN243</td>
      <td>R/V Thompson</td>
      <td>2009-11-07T23:07:30.000Z</td>
      <td>2009-11-09T20:17:42.000Z</td>
      <td>47.296100</td>
      <td>48.375200</td>
      <td>-122.823500</td>
      <td>-122.332700</td>
      <td>Virginia Armbrust</td>
      <td>2009, Armbrust, California, CCAL, CMAP, Coasta...</td>
    </tr>
    <tr>
      <th>67</th>
      <td>555</td>
      <td>Thompson_1</td>
      <td>TN248</td>
      <td>R/V Thompson</td>
      <td>2010-05-04T23:00:34.000Z</td>
      <td>2010-05-08T14:12:05.000Z</td>
      <td>49.937300</td>
      <td>54.150200</td>
      <td>-166.495900</td>
      <td>-136.272900</td>
      <td>NaN</td>
      <td>(East), 2010, Alaska, ALSK, Armbrust, chroco, ...</td>
    </tr>
    <tr>
      <th>68</th>
      <td>556</td>
      <td>Thompson_2</td>
      <td>TN250</td>
      <td>R/V Thompson</td>
      <td>2010-06-21T00:00:44.000Z</td>
      <td>2010-07-07T23:59:50.000Z</td>
      <td>55.011800</td>
      <td>62.210000</td>
      <td>-178.969300</td>
      <td>-165.866000</td>
      <td>David Shull</td>
      <td>2010, BERS, CMAP, cruises, data, David, Epicon...</td>
    </tr>
    <tr>
      <th>69</th>
      <td>557</td>
      <td>Thompson_3</td>
      <td>TN252</td>
      <td>R/V Thompson</td>
      <td>2010-07-27T04:07:47.000Z</td>
      <td>2010-08-23T10:41:43.000Z</td>
      <td>44.341600</td>
      <td>48.514200</td>
      <td>-130.080800</td>
      <td>-122.590200</td>
      <td>John Delaney</td>
      <td>2010, Aug, August, California, CCAL, CMAP, Coa...</td>
    </tr>
    <tr>
      <th>70</th>
      <td>558</td>
      <td>Thompson_4</td>
      <td>TN257</td>
      <td>R/V Thompson</td>
      <td>2010-11-10T01:07:09.000Z</td>
      <td>2010-11-18T03:34:54.000Z</td>
      <td>21.225800</td>
      <td>48.505500</td>
      <td>-157.876700</td>
      <td>-122.539800</td>
      <td>NaN</td>
      <td>2010, California, CCAL, CMAP, Coastal, cruises...</td>
    </tr>
    <tr>
      <th>71</th>
      <td>559</td>
      <td>Thompson_5</td>
      <td>TN260</td>
      <td>R/V Thompson</td>
      <td>2010-12-27T23:30:48.000Z</td>
      <td>2011-01-04T11:01:02.000Z</td>
      <td>18.658500</td>
      <td>21.339900</td>
      <td>-158.764100</td>
      <td>-155.226700</td>
      <td>Richard Keil</td>
      <td>2010, 2011, CMAP, cruises, data, Dec, December...</td>
    </tr>
    <tr>
      <th>72</th>
      <td>560</td>
      <td>Thompson_6</td>
      <td>TN264-265</td>
      <td>R/V Thompson</td>
      <td>2011-05-22T01:41:45.000Z</td>
      <td>2011-06-24T05:36:36.000Z</td>
      <td>40.256400</td>
      <td>49.069500</td>
      <td>-126.864000</td>
      <td>-122.514300</td>
      <td>Matthew Alford, Christopher Romsos</td>
      <td>2011, Alford,, California, CCAL, Christopher, ...</td>
    </tr>
    <tr>
      <th>73</th>
      <td>563</td>
      <td>Thompson_7</td>
      <td>TN268</td>
      <td>R/V Thompson</td>
      <td>2011-08-21T04:20:35.000Z</td>
      <td>2011-08-31T23:58:43.000Z</td>
      <td>45.437800</td>
      <td>48.473700</td>
      <td>-130.093800</td>
      <td>-123.760100</td>
      <td>John Delaney</td>
      <td>2011, Aug, August, California, CCAL, CMAP, Coa...</td>
    </tr>
    <tr>
      <th>74</th>
      <td>565</td>
      <td>Thompson_8</td>
      <td>TN270</td>
      <td>R/V Thompson</td>
      <td>2011-10-08T20:53:23.000Z</td>
      <td>2011-10-10T19:58:01.000Z</td>
      <td>47.119200</td>
      <td>47.703600</td>
      <td>-122.853300</td>
      <td>-122.348900</td>
      <td>Multiple - Virginia Armbrust</td>
      <td>-, 2011, Armbrust, California, CCAL, CMAP, Coa...</td>
    </tr>
    <tr>
      <th>75</th>
      <td>566</td>
      <td>Thompson_9</td>
      <td>TN271</td>
      <td>R/V Thompson</td>
      <td>2011-10-26T04:34:38.000Z</td>
      <td>2011-11-03T00:48:24.000Z</td>
      <td>21.229000</td>
      <td>48.512800</td>
      <td>-157.882100</td>
      <td>-124.250500</td>
      <td>Francois Ribalet</td>
      <td>2011, Armbrust, California, CCAL, chroco, CMAP...</td>
    </tr>
    <tr>
      <th>76</th>
      <td>569</td>
      <td>Thompson_10</td>
      <td>TN280</td>
      <td>R/V Thompson</td>
      <td>2012-05-16T16:23:07.000Z</td>
      <td>2012-05-22T12:04:56.000Z</td>
      <td>47.678300</td>
      <td>48.816900</td>
      <td>-128.666900</td>
      <td>-122.414900</td>
      <td>Virginia Armbrust</td>
      <td>2012, Armbrust, California, CCAL, chroco, CMAP...</td>
    </tr>
    <tr>
      <th>77</th>
      <td>571</td>
      <td>Thompson_11</td>
      <td>TN291</td>
      <td>R/V Thompson</td>
      <td>2013-01-26T23:14:28.000Z</td>
      <td>2013-02-02T16:07:38.000Z</td>
      <td>47.928300</td>
      <td>49.221700</td>
      <td>-126.105100</td>
      <td>-122.595600</td>
      <td>Richard Keil</td>
      <td>2013, California, CCAL, CMAP, Coastal, cruises...</td>
    </tr>
    <tr>
      <th>78</th>
      <td>572</td>
      <td>Thompson_12</td>
      <td>TN292</td>
      <td>R/V Thompson</td>
      <td>2013-03-05T22:38:45.000Z</td>
      <td>2013-03-13T02:50:56.000Z</td>
      <td>24.633822</td>
      <td>48.511483</td>
      <td>-154.021572</td>
      <td>-122.445282</td>
      <td>NaN</td>
      <td>2013, Armbrust, California, CCAL, chroco, CMAP...</td>
    </tr>
    <tr>
      <th>79</th>
      <td>561</td>
      <td>Tokyo_1</td>
      <td>Tokyo_1</td>
      <td>M/V Tokyo</td>
      <td>2011-05-16T15:01:58.000Z</td>
      <td>2011-05-30T03:15:35.000Z</td>
      <td>22.920600</td>
      <td>43.329300</td>
      <td>-179.984800</td>
      <td>179.991300</td>
      <td>NaN</td>
      <td>(West), 2011, California, CCAL, CHIN, China, C...</td>
    </tr>
    <tr>
      <th>80</th>
      <td>562</td>
      <td>Tokyo_2</td>
      <td>Tokyo_2</td>
      <td>M/V Tokyo</td>
      <td>2011-06-27T06:57:50.000Z</td>
      <td>2011-07-10T23:12:13.000Z</td>
      <td>22.123100</td>
      <td>49.568600</td>
      <td>-179.990300</td>
      <td>179.988700</td>
      <td>NaN</td>
      <td>(East), (West), 2011, BERS, California, CCAL, ...</td>
    </tr>
    <tr>
      <th>81</th>
      <td>567</td>
      <td>Tokyo_3</td>
      <td>Tokyo_3</td>
      <td>M/V Tokyo</td>
      <td>2011-09-19T04:40:53.000Z</td>
      <td>2011-10-02T20:53:05.000Z</td>
      <td>22.105100</td>
      <td>44.592400</td>
      <td>-179.993800</td>
      <td>179.982900</td>
      <td>NaN</td>
      <td>(East), (West), 2011, California, CCAL, CHIN, ...</td>
    </tr>
    <tr>
      <th>82</th>
      <td>568</td>
      <td>Tokyo_4</td>
      <td>Tokyo_4</td>
      <td>M/V Tokyo</td>
      <td>2012-01-31T04:08:23.000Z</td>
      <td>2012-02-06T21:04:53.000Z</td>
      <td>33.628700</td>
      <td>40.307800</td>
      <td>-179.984700</td>
      <td>179.992000</td>
      <td>NaN</td>
      <td>2012, California, CCAL, CMAP, Coastal, cruises...</td>
    </tr>
  </tbody>
</table>
<p>83 rows × 12 columns</p>
</div></div>
</div>
</div>
<div class="section" id="Example:">
<h2>Example:<a class="headerlink" href="#Example:" title="Permalink to this headline">¶</a></h2>
<p>A sample query returning the timeseries of sea surface temperature (sst).</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">api</span><span class="o">.</span><span class="n">query</span><span class="p">(</span>
         <span class="sd">&#39;&#39;&#39;</span>
<span class="sd">         SELECT [time], AVG(lat) AS lat, AVG(lon) AS lon, AVG(sst) AS sst FROM tblsst_AVHRR_OI_NRT</span>
<span class="sd">         WHERE</span>
<span class="sd">         [time] BETWEEN &#39;2016-06-01&#39; AND &#39;2016-10-01&#39; AND</span>
<span class="sd">         lat BETWEEN 23 AND 24 AND</span>
<span class="sd">         lon BETWEEN -160 AND -158</span>
<span class="sd">         GROUP BY [time]</span>
<span class="sd">         ORDER BY [time]</span>
<span class="sd">         &#39;&#39;&#39;</span>
         <span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>time</th>
      <th>lat</th>
      <th>lon</th>
      <th>sst</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>2016-06-01T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.316556</td>
    </tr>
    <tr>
      <th>1</th>
      <td>2016-06-02T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.277493</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2016-06-03T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.245618</td>
    </tr>
    <tr>
      <th>3</th>
      <td>2016-06-04T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.157493</td>
    </tr>
    <tr>
      <th>4</th>
      <td>2016-06-05T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.301243</td>
    </tr>
    <tr>
      <th>5</th>
      <td>2016-06-06T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.269056</td>
    </tr>
    <tr>
      <th>6</th>
      <td>2016-06-07T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.404056</td>
    </tr>
    <tr>
      <th>7</th>
      <td>2016-06-08T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.521868</td>
    </tr>
    <tr>
      <th>8</th>
      <td>2016-06-09T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.552493</td>
    </tr>
    <tr>
      <th>9</th>
      <td>2016-06-10T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.403743</td>
    </tr>
    <tr>
      <th>10</th>
      <td>2016-06-11T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.417806</td>
    </tr>
    <tr>
      <th>11</th>
      <td>2016-06-12T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.573118</td>
    </tr>
    <tr>
      <th>12</th>
      <td>2016-06-13T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.730306</td>
    </tr>
    <tr>
      <th>13</th>
      <td>2016-06-14T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.996556</td>
    </tr>
    <tr>
      <th>14</th>
      <td>2016-06-15T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.926243</td>
    </tr>
    <tr>
      <th>15</th>
      <td>2016-06-16T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.061556</td>
    </tr>
    <tr>
      <th>16</th>
      <td>2016-06-17T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.029681</td>
    </tr>
    <tr>
      <th>17</th>
      <td>2016-06-18T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.809056</td>
    </tr>
    <tr>
      <th>18</th>
      <td>2016-06-19T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.748118</td>
    </tr>
    <tr>
      <th>19</th>
      <td>2016-06-20T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.950306</td>
    </tr>
    <tr>
      <th>20</th>
      <td>2016-06-21T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.574993</td>
    </tr>
    <tr>
      <th>21</th>
      <td>2016-06-22T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.444681</td>
    </tr>
    <tr>
      <th>22</th>
      <td>2016-06-23T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.994993</td>
    </tr>
    <tr>
      <th>23</th>
      <td>2016-06-24T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.122493</td>
    </tr>
    <tr>
      <th>24</th>
      <td>2016-06-25T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.069681</td>
    </tr>
    <tr>
      <th>25</th>
      <td>2016-06-26T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.929681</td>
    </tr>
    <tr>
      <th>26</th>
      <td>2016-06-27T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.975618</td>
    </tr>
    <tr>
      <th>27</th>
      <td>2016-06-28T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.818743</td>
    </tr>
    <tr>
      <th>28</th>
      <td>2016-06-29T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.481556</td>
    </tr>
    <tr>
      <th>29</th>
      <td>2016-06-30T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>25.635618</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>93</th>
      <td>2016-09-02T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.678431</td>
    </tr>
    <tr>
      <th>94</th>
      <td>2016-09-03T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.726243</td>
    </tr>
    <tr>
      <th>95</th>
      <td>2016-09-04T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.761243</td>
    </tr>
    <tr>
      <th>96</th>
      <td>2016-09-05T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.759681</td>
    </tr>
    <tr>
      <th>97</th>
      <td>2016-09-06T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.727806</td>
    </tr>
    <tr>
      <th>98</th>
      <td>2016-09-07T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.676868</td>
    </tr>
    <tr>
      <th>99</th>
      <td>2016-09-08T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.635618</td>
    </tr>
    <tr>
      <th>100</th>
      <td>2016-09-09T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.674056</td>
    </tr>
    <tr>
      <th>101</th>
      <td>2016-09-10T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.745306</td>
    </tr>
    <tr>
      <th>102</th>
      <td>2016-09-11T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.739681</td>
    </tr>
    <tr>
      <th>103</th>
      <td>2016-09-12T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.697806</td>
    </tr>
    <tr>
      <th>104</th>
      <td>2016-09-13T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.679993</td>
    </tr>
    <tr>
      <th>105</th>
      <td>2016-09-14T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.675306</td>
    </tr>
    <tr>
      <th>106</th>
      <td>2016-09-15T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.505618</td>
    </tr>
    <tr>
      <th>107</th>
      <td>2016-09-16T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.480618</td>
    </tr>
    <tr>
      <th>108</th>
      <td>2016-09-17T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.620618</td>
    </tr>
    <tr>
      <th>109</th>
      <td>2016-09-18T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.817493</td>
    </tr>
    <tr>
      <th>110</th>
      <td>2016-09-19T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.791868</td>
    </tr>
    <tr>
      <th>111</th>
      <td>2016-09-20T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.515931</td>
    </tr>
    <tr>
      <th>112</th>
      <td>2016-09-21T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.394056</td>
    </tr>
    <tr>
      <th>113</th>
      <td>2016-09-22T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.137181</td>
    </tr>
    <tr>
      <th>114</th>
      <td>2016-09-23T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.128431</td>
    </tr>
    <tr>
      <th>115</th>
      <td>2016-09-24T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.353431</td>
    </tr>
    <tr>
      <th>116</th>
      <td>2016-09-25T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.436556</td>
    </tr>
    <tr>
      <th>117</th>
      <td>2016-09-26T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.480931</td>
    </tr>
    <tr>
      <th>118</th>
      <td>2016-09-27T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.449993</td>
    </tr>
    <tr>
      <th>119</th>
      <td>2016-09-28T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.394056</td>
    </tr>
    <tr>
      <th>120</th>
      <td>2016-09-29T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.378118</td>
    </tr>
    <tr>
      <th>121</th>
      <td>2016-09-30T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.425618</td>
    </tr>
    <tr>
      <th>122</th>
      <td>2016-10-01T00:00:00</td>
      <td>23.5</td>
      <td>-159</td>
      <td>26.346868</td>
    </tr>
  </tbody>
</table>
<p>123 rows × 4 columns</p>
</div></div>
</div>
<center><h1><p>Study Cases</p>
</h1>  </center>






Table of Contents ## (see slides →) ## Case Study 1: Attach Environmental Parameters to the SeaFlow Observations<p>In this study, we take all seaflow cruises (approximately 35 cruises) and colocalize them with 50+ environmental variables. The idea is to identify the highly-correlated environmental variables (correlated with seaflow abundances). These variables then serve as predictors of machine learning algorithms that capture the seaflow variations. The trained machine learning models then are used to generate spatial maps of pico-phytoplanktons (Prochlorococcus, Synechococcus, and pico-Eukaryotes).</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">Author: Mohammad Dehghani Ashkezari &lt;mdehghan@uw.edu&gt;</span>

<span class="sd">Date: 2019-08-13</span>

<span class="sd">Function: Colocalizes tens of variables along-track of cruises with underway Seaflow measurements.</span>
<span class="sd">&quot;&quot;&quot;</span>

<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">pycmap</span>
<span class="kn">from</span> <span class="nn">collections</span> <span class="k">import</span> <span class="n">namedtuple</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>


<span class="k">def</span> <span class="nf">all_cruises</span><span class="p">(</span><span class="n">api</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Returns a list of seaflow cruises, excluding the AMT cruises.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">cruises</span> <span class="o">=</span> <span class="n">api</span><span class="o">.</span><span class="n">cruises</span><span class="p">()</span><span class="o">.</span><span class="n">Name</span>
    <span class="k">return</span> <span class="nb">list</span><span class="p">(</span><span class="n">cruises</span><span class="p">[</span><span class="o">~</span><span class="n">cruises</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">contains</span><span class="p">(</span><span class="s2">&quot;AMT&quot;</span><span class="p">)])</span>


<span class="k">def</span> <span class="nf">match_params</span><span class="p">():</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Creates a collection of variables (and their tolerances) to be colocalized along the cruise trajectory.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">Param</span> <span class="o">=</span> <span class="n">namedtuple</span><span class="p">(</span><span class="s1">&#39;Param&#39;</span><span class="p">,</span> <span class="p">[</span><span class="s1">&#39;table&#39;</span><span class="p">,</span> <span class="s1">&#39;variable&#39;</span><span class="p">,</span> <span class="s1">&#39;temporalTolerance&#39;</span><span class="p">,</span> <span class="s1">&#39;latTolerance&#39;</span><span class="p">,</span> <span class="s1">&#39;lonTolerance&#39;</span><span class="p">,</span> <span class="s1">&#39;depthTolerance&#39;</span><span class="p">])</span>
    <span class="n">params</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblSeaFlow&#39;</span><span class="p">,</span> <span class="s1">&#39;prochloro_abundance&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblSeaFlow&#39;</span><span class="p">,</span> <span class="s1">&#39;prochloro_diameter&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblSeaFlow&#39;</span><span class="p">,</span> <span class="s1">&#39;prochloro_carbon_content&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblSeaFlow&#39;</span><span class="p">,</span> <span class="s1">&#39;prochloro_biomass&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblSeaFlow&#39;</span><span class="p">,</span> <span class="s1">&#39;synecho_abundance&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblSeaFlow&#39;</span><span class="p">,</span> <span class="s1">&#39;synecho_diameter&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblSeaFlow&#39;</span><span class="p">,</span> <span class="s1">&#39;synecho_carbon_content&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblSeaFlow&#39;</span><span class="p">,</span> <span class="s1">&#39;synecho_biomass&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblSeaFlow&#39;</span><span class="p">,</span> <span class="s1">&#39;picoeuk_abundance&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblSeaFlow&#39;</span><span class="p">,</span> <span class="s1">&#39;picoeuk_diameter&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblSeaFlow&#39;</span><span class="p">,</span> <span class="s1">&#39;picoeuk_carbon_content&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblSeaFlow&#39;</span><span class="p">,</span> <span class="s1">&#39;picoeuk_biomass&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblSeaFlow&#39;</span><span class="p">,</span> <span class="s1">&#39;total_biomass&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>

    <span class="c1">######## Ship Data (not calibrated)</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblCruise_Salinity&#39;</span><span class="p">,</span> <span class="s1">&#39;salinity&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblCruise_Temperature&#39;</span><span class="p">,</span> <span class="s1">&#39;temperature&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mf">0.1</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>

    <span class="c1">######## satellite</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblSST_AVHRR_OI_NRT&#39;</span><span class="p">,</span> <span class="s1">&#39;sst&#39;</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mf">0.25</span><span class="p">,</span> <span class="mf">0.25</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblSSS_NRT&#39;</span><span class="p">,</span> <span class="s1">&#39;sss&#39;</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mf">0.25</span><span class="p">,</span> <span class="mf">0.25</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblCHL_REP&#39;</span><span class="p">,</span> <span class="s1">&#39;chl&#39;</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mf">0.25</span><span class="p">,</span> <span class="mf">0.25</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblModis_AOD_REP&#39;</span><span class="p">,</span> <span class="s1">&#39;AOD&#39;</span><span class="p">,</span> <span class="mi">15</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblAltimetry_REP&#39;</span><span class="p">,</span> <span class="s1">&#39;sla&#39;</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mf">0.25</span><span class="p">,</span> <span class="mf">0.25</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblAltimetry_REP&#39;</span><span class="p">,</span> <span class="s1">&#39;adt&#39;</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mf">0.25</span><span class="p">,</span> <span class="mf">0.25</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblAltimetry_REP&#39;</span><span class="p">,</span> <span class="s1">&#39;ugos&#39;</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mf">0.25</span><span class="p">,</span> <span class="mf">0.25</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblAltimetry_REP&#39;</span><span class="p">,</span> <span class="s1">&#39;vgos&#39;</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mf">0.25</span><span class="p">,</span> <span class="mf">0.25</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>

    <span class="c1">######## model</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblPisces_NRT&#39;</span><span class="p">,</span> <span class="s1">&#39;Fe&#39;</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblPisces_NRT&#39;</span><span class="p">,</span> <span class="s1">&#39;NO3&#39;</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblPisces_NRT&#39;</span><span class="p">,</span> <span class="s1">&#39;O2&#39;</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblPisces_NRT&#39;</span><span class="p">,</span> <span class="s1">&#39;PO4&#39;</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblPisces_NRT&#39;</span><span class="p">,</span> <span class="s1">&#39;Si&#39;</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblPisces_NRT&#39;</span><span class="p">,</span> <span class="s1">&#39;PP&#39;</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblPisces_NRT&#39;</span><span class="p">,</span> <span class="s1">&#39;CHL&#39;</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblDarwin_Nutrient_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;NH4_darwin_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblDarwin_Nutrient_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;NO2_darwin_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblDarwin_Nutrient_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;SiO2_darwin_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblDarwin_Nutrient_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;DOC_darwin_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblDarwin_Nutrient_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;DON_darwin_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblDarwin_Nutrient_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;DOP_darwin_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblDarwin_Nutrient_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;DOFe_darwin_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblDarwin_Nutrient_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;PIC_darwin_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblDarwin_Nutrient_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;ALK_darwin_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblDarwin_Nutrient_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;FeT_darwin_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblDarwin_Plankton_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;prokaryote_c01_darwin_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblDarwin_Plankton_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;prokaryote_c02_darwin_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblDarwin_Plankton_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;picoeukaryote_c03_darwin_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblDarwin_Plankton_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;picoeukaryote_c04_darwin_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>

    <span class="c1">####### World Ocean Atlas (WOA)</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblWOA_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;density_WOA_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="o">.</span><span class="mi">75</span><span class="p">,</span> <span class="o">.</span><span class="mi">75</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblWOA_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;nitrate_WOA_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.75</span><span class="p">,</span> <span class="mf">0.75</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblWOA_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;phosphate_WOA_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.75</span><span class="p">,</span> <span class="mf">0.75</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblWOA_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;silicate_WOA_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.75</span><span class="p">,</span> <span class="mf">0.75</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblWOA_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;oxygen_WOA_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.75</span><span class="p">,</span> <span class="mf">0.75</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblWOA_Climatology&#39;</span><span class="p">,</span> <span class="s1">&#39;salinity_WOA_clim&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.75</span><span class="p">,</span> <span class="mf">0.75</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>

    <span class="n">tables</span><span class="p">,</span> <span class="n">variables</span><span class="p">,</span> <span class="n">temporalTolerance</span><span class="p">,</span> <span class="n">latTolerance</span><span class="p">,</span> <span class="n">lonTolerance</span><span class="p">,</span> <span class="n">depthTolerance</span> <span class="o">=</span> <span class="p">[],</span> <span class="p">[],</span> <span class="p">[],</span> <span class="p">[],</span> <span class="p">[],</span> <span class="p">[]</span>
    <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">params</span><span class="p">)):</span>
        <span class="n">tables</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">params</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">table</span><span class="p">)</span>
        <span class="n">variables</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">params</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">variable</span><span class="p">)</span>
        <span class="n">temporalTolerance</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">params</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">temporalTolerance</span><span class="p">)</span>
        <span class="n">latTolerance</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">params</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">latTolerance</span><span class="p">)</span>
        <span class="n">lonTolerance</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">params</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">lonTolerance</span><span class="p">)</span>
        <span class="n">depthTolerance</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">params</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">depthTolerance</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">tables</span><span class="p">,</span> <span class="n">variables</span><span class="p">,</span> <span class="n">temporalTolerance</span><span class="p">,</span> <span class="n">latTolerance</span><span class="p">,</span> <span class="n">lonTolerance</span><span class="p">,</span> <span class="n">depthTolerance</span>




<span class="k">def</span> <span class="nf">main</span><span class="p">():</span>
    <span class="n">api</span> <span class="o">=</span> <span class="n">pycmap</span><span class="o">.</span><span class="n">API</span><span class="p">()</span>
    <span class="n">cruises</span> <span class="o">=</span> <span class="n">all_cruises</span><span class="p">(</span><span class="n">api</span><span class="p">)</span>
    <span class="n">cruises</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;KOK1606&#39;</span><span class="p">]</span>   <span class="c1"># limiting to only one cruise (for presentation)</span>
    <span class="n">exportDir</span> <span class="o">=</span> <span class="s1">&#39;./export/&#39;</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">exists</span><span class="p">(</span><span class="n">exportDir</span><span class="p">):</span> <span class="n">os</span><span class="o">.</span><span class="n">makedirs</span><span class="p">(</span><span class="n">exportDir</span><span class="p">)</span>
    <span class="n">tables</span><span class="p">,</span> <span class="n">variables</span><span class="p">,</span> <span class="n">temporalTolerance</span><span class="p">,</span> <span class="n">latTolerance</span><span class="p">,</span> <span class="n">lonTolerance</span><span class="p">,</span> <span class="n">depthTolerance</span> <span class="o">=</span> <span class="n">match_params</span><span class="p">()</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">({})</span>
    <span class="k">for</span> <span class="n">cruise</span> <span class="ow">in</span> <span class="n">cruises</span><span class="p">:</span>
        <span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\n</span><span class="s1">********************************&#39;</span><span class="p">)</span>
        <span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Preparing </span><span class="si">%s</span><span class="s1"> cruise...&#39;</span> <span class="o">%</span> <span class="n">cruise</span><span class="p">)</span>
        <span class="nb">print</span><span class="p">(</span><span class="s1">&#39;********************************</span><span class="se">\n</span><span class="s1">&#39;</span><span class="p">)</span>
        <span class="n">data</span> <span class="o">=</span> <span class="n">api</span><span class="o">.</span><span class="n">along_track</span><span class="p">(</span>
                              <span class="n">cruise</span><span class="o">=</span><span class="n">cruise</span><span class="p">,</span>
                              <span class="n">targetTables</span><span class="o">=</span><span class="n">tables</span><span class="p">,</span>
                              <span class="n">targetVars</span><span class="o">=</span><span class="n">variables</span><span class="p">,</span>
                              <span class="n">temporalTolerance</span><span class="o">=</span><span class="n">temporalTolerance</span><span class="p">,</span>
                              <span class="n">latTolerance</span><span class="o">=</span><span class="n">latTolerance</span><span class="p">,</span>
                              <span class="n">lonTolerance</span><span class="o">=</span><span class="n">lonTolerance</span><span class="p">,</span>
                              <span class="n">depthTolerance</span><span class="o">=</span><span class="n">depthTolerance</span><span class="p">,</span>
                              <span class="n">depth1</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span>
                              <span class="n">depth2</span><span class="o">=</span><span class="mi">5</span>
                              <span class="p">)</span>
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">df</span><span class="p">)</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
            <span class="n">df</span> <span class="o">=</span> <span class="n">data</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">([</span><span class="n">df</span><span class="p">,</span> <span class="n">data</span><span class="p">],</span> <span class="n">ignore_index</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
        <span class="n">data</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%s%s</span><span class="s1">.csv&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">exportDir</span><span class="p">,</span> <span class="n">cruise</span><span class="p">),</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
    <span class="n">df</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%s</span><span class="s1">sfMatch.csv&#39;</span> <span class="o">%</span> <span class="n">exportDir</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">df</span>





<span class="c1">##############################</span>
<span class="c1">#                            #</span>
<span class="c1">#           main             #</span>
<span class="c1">#                            #</span>
<span class="c1">##############################</span>


<span class="k">if</span> <span class="vm">__name__</span> <span class="o">==</span> <span class="s1">&#39;__main__&#39;</span><span class="p">:</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">main</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>

********************************
Preparing KOK1606 cruise...
********************************

</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<script type="application/vnd.jupyter.widget-view+json">{"model_id": "3bd4da5143ca4f0aacef70b437dbe7f4", "version_major": 2, "version_minor": 0}</script></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<blockquote>
<div><div class="highlight"><pre>

1: prochloro_abundance matched.

2: prochloro_diameter matched.

3: prochloro_carbon_content matched.

4: prochloro_biomass matched.

5: synecho_abundance matched.

6: synecho_diameter matched.

7: synecho_carbon_content matched.

8: synecho_biomass matched.

9: picoeuk_abundance matched.

10: picoeuk_diameter matched.

11: picoeuk_carbon_content matched.

12: picoeuk_biomass matched.

13: total_biomass matched.

14: salinity matched.

15: temperature matched.

16: sst matched.

17: sss matched.

18: chl matched.

19: AOD matched.

20: sla matched.

21: adt matched.

22: ugos matched.

23: vgos matched.

24: Fe matched.

25: NO3 matched.

26: O2 matched.

27: PO4 matched.

28: Si matched.

29: PP matched.

30: CHL matched.

31: NH4_darwin_clim matched.

32: NO2_darwin_clim matched.

33: SiO2_darwin_clim matched.

34: DOC_darwin_clim matched.

35: DON_darwin_clim matched.

36: DOP_darwin_clim matched.

37: DOFe_darwin_clim matched.

38: PIC_darwin_clim matched.

39: ALK_darwin_clim matched.

40: FeT_darwin_clim matched.

41: prokaryote_c01_darwin_clim matched.

42: prokaryote_c02_darwin_clim matched.

43: picoeukaryote_c03_darwin_clim matched.

44: picoeukaryote_c04_darwin_clim matched.

45: density_WOA_clim matched.

46: nitrate_WOA_clim matched.

47: phosphate_WOA_clim matched.

48: silicate_WOA_clim matched.

49: oxygen_WOA_clim matched.

50: salinity_WOA_clim matched.

</pre></div></div></blockquote>
<p>Table of Contents ## (see slides →) ## Case Study 2: Inter-Annual Variability of Eddy Induced Temperature Anomaly In this example, we iteratively retrieve daily eddy locations and colocalize them with satellite and model variables (SST, CHL, SLA, and NO3). To infer the eddy induced effects we also compute an estimate of the local background. Subtracting the background field from that of eddy domain results in the eddy induced effects. For demonstration purposes, the script below is limited to a</p>
</div>
</div>
<p>small region within a one-day period (see the root of the script).</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="sd">&quot;&quot;&quot;</span>
<span class="sd">Author: Mohammad Dehghani Ashkezari &lt;mdehghan@uw.edu&gt;</span>

<span class="sd">Date: 2019-11-01</span>

<span class="sd">Function: Colocalize (match) eddy data set with a number of satellite &amp; model variables (e.g. SST, CHL, NO3, etc ...).</span>
<span class="sd">&quot;&quot;&quot;</span>

<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">pycmap</span>
<span class="kn">from</span> <span class="nn">collections</span> <span class="k">import</span> <span class="n">namedtuple</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">from</span> <span class="nn">datetime</span> <span class="k">import</span> <span class="n">datetime</span><span class="p">,</span> <span class="n">timedelta</span><span class="p">,</span> <span class="n">date</span>




<span class="k">def</span> <span class="nf">sparse_dates</span><span class="p">(</span><span class="n">y1</span><span class="p">,</span> <span class="n">y2</span><span class="p">,</span> <span class="n">m1</span><span class="p">,</span> <span class="n">m2</span><span class="p">,</span> <span class="n">d1</span><span class="p">,</span> <span class="n">d2</span><span class="p">):</span>
    <span class="n">dts</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="k">for</span> <span class="n">y</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">y1</span><span class="p">,</span> <span class="n">y2</span><span class="o">+</span><span class="mi">1</span><span class="p">):</span>
        <span class="k">for</span> <span class="n">m</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">m1</span><span class="p">,</span> <span class="n">m2</span><span class="o">+</span><span class="mi">1</span><span class="p">):</span>
            <span class="k">for</span> <span class="n">d</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">d1</span><span class="p">,</span> <span class="n">d2</span><span class="o">+</span><span class="mi">1</span><span class="p">):</span>
                <span class="n">dts</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">datetime</span><span class="p">(</span><span class="n">y</span><span class="p">,</span> <span class="n">m</span><span class="p">,</span> <span class="n">d</span><span class="p">))</span>
    <span class="k">return</span> <span class="n">dts</span>


<span class="k">def</span> <span class="nf">eddy_time_range</span><span class="p">(</span><span class="n">api</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Returns the start-date and end-date of the eddy dataset.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">query</span> <span class="o">=</span> <span class="s2">&quot;SELECT min([time]) AS min_time, max([time]) max_time FROM tblMesoscale_Eddy&quot;</span>
    <span class="n">df</span> <span class="o">=</span> <span class="n">api</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="n">query</span><span class="p">)</span>
    <span class="n">dt1</span> <span class="o">=</span> <span class="n">datetime</span><span class="o">.</span><span class="n">strptime</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="s1">&#39;min_time&#39;</span><span class="p">],</span> <span class="s1">&#39;%Y-%m-</span><span class="si">%d</span><span class="s1">T%H:%M:%S.000Z&#39;</span><span class="p">)</span>
    <span class="n">dt2</span> <span class="o">=</span> <span class="n">datetime</span><span class="o">.</span><span class="n">strptime</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="s1">&#39;max_time&#39;</span><span class="p">],</span> <span class="s1">&#39;%Y-%m-</span><span class="si">%d</span><span class="s1">T%H:%M:%S.000Z&#39;</span><span class="p">)</span>
    <span class="k">return</span> <span class="p">[</span><span class="n">dt1</span> <span class="o">+</span> <span class="n">timedelta</span><span class="p">(</span><span class="n">days</span><span class="o">=</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">((</span><span class="n">dt2</span><span class="o">-</span><span class="n">dt1</span><span class="p">)</span><span class="o">.</span><span class="n">days</span> <span class="o">+</span> <span class="mi">1</span><span class="p">)]</span>


<span class="k">def</span> <span class="nf">daily_eddies</span><span class="p">(</span><span class="n">api</span><span class="p">,</span> <span class="n">day</span><span class="p">,</span> <span class="n">lat1</span><span class="p">,</span> <span class="n">lat2</span><span class="p">,</span> <span class="n">lon1</span><span class="p">,</span> <span class="n">lon2</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Returns eddies at a given date (day) delimited by the spatial parameters (lat1, lat2, lon1, lon2).</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">query</span> <span class="o">=</span> <span class="s2">&quot;&quot;&quot;</span>
<span class="s2">            SELECT * FROM tblMesoscale_Eddy</span>
<span class="s2">            WHERE</span>
<span class="s2">            [time]=&#39;</span><span class="si">%s</span><span class="s2">&#39;</span>
<span class="s2">            AND</span>
<span class="s2">            lat BETWEEN </span><span class="si">%f</span><span class="s2"> AND </span><span class="si">%f</span><span class="s2"> AND</span>
<span class="s2">            lon BETWEEN </span><span class="si">%f</span><span class="s2"> AND </span><span class="si">%f</span><span class="s2"></span>
<span class="s2">            &quot;&quot;&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">day</span><span class="p">,</span> <span class="n">lat1</span><span class="p">,</span> <span class="n">lat2</span><span class="p">,</span> <span class="n">lon1</span><span class="p">,</span> <span class="n">lon2</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">api</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="n">query</span><span class="p">)</span>


<span class="k">def</span> <span class="nf">match_covariate</span><span class="p">(</span><span class="n">api</span><span class="p">,</span> <span class="n">table</span><span class="p">,</span> <span class="n">variable</span><span class="p">,</span> <span class="n">dt1</span><span class="p">,</span> <span class="n">dt2</span><span class="p">,</span> <span class="n">lat</span><span class="p">,</span> <span class="n">del_lat</span><span class="p">,</span> <span class="n">lon</span><span class="p">,</span> <span class="n">del_lon</span><span class="p">,</span> <span class="n">depth</span><span class="p">,</span> <span class="n">del_depth</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Returns the mean and standard-deviation of variable within the eddy domain and with the background field.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="k">def</span> <span class="nf">has_depth</span><span class="p">(</span><span class="n">table</span><span class="p">):</span>
        <span class="k">return</span> <span class="n">table</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;tblPisces_NRT&#39;</span><span class="p">,</span> <span class="s1">&#39;tblDarwin_Nutrient&#39;</span><span class="p">,</span> <span class="s1">&#39;tblDarwin_Ecosystem&#39;</span><span class="p">,</span> <span class="s1">&#39;tblDarwin_Phytoplankton&#39;</span><span class="p">]</span>
    <span class="n">query</span> <span class="o">=</span> <span class="s2">&quot;SELECT AVG(</span><span class="si">%s</span><span class="s2">) AS </span><span class="si">%s</span><span class="s2">, STDEV(</span><span class="si">%s</span><span class="s2">) AS </span><span class="si">%s</span><span class="s2"> FROM </span><span class="si">%s</span><span class="s2"> &quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">variable</span><span class="p">,</span> <span class="n">variable</span><span class="p">,</span> <span class="n">variable</span><span class="p">,</span> <span class="n">variable</span><span class="o">+</span><span class="s1">&#39;_std&#39;</span><span class="p">,</span> <span class="n">table</span><span class="p">)</span>
    <span class="n">query</span> <span class="o">+=</span> <span class="s2">&quot;WHERE [time] BETWEEN &#39;</span><span class="si">%s</span><span class="s2">&#39; AND &#39;</span><span class="si">%s</span><span class="s2">&#39; AND &quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">dt1</span><span class="p">,</span> <span class="n">dt2</span><span class="p">)</span>
    <span class="n">query</span> <span class="o">+=</span> <span class="s2">&quot;[lat] BETWEEN </span><span class="si">%f</span><span class="s2"> AND </span><span class="si">%f</span><span class="s2"> AND &quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">lat</span><span class="o">-</span><span class="n">del_lat</span><span class="p">,</span> <span class="n">lat</span><span class="o">+</span><span class="n">del_lat</span><span class="p">)</span>
    <span class="n">query</span> <span class="o">+=</span> <span class="s2">&quot;[lon] BETWEEN </span><span class="si">%f</span><span class="s2"> AND </span><span class="si">%f</span><span class="s2"> &quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">lon</span><span class="o">-</span><span class="n">del_lon</span><span class="p">,</span> <span class="n">lon</span><span class="o">+</span><span class="n">del_lon</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">has_depth</span><span class="p">(</span><span class="n">table</span><span class="p">):</span>
        <span class="n">query</span> <span class="o">+=</span> <span class="s2">&quot; AND [depth] BETWEEN </span><span class="si">%f</span><span class="s2"> AND </span><span class="si">%f</span><span class="s2"> &quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">depth</span><span class="o">-</span><span class="n">del_depth</span><span class="p">,</span> <span class="n">depth</span><span class="o">+</span><span class="n">del_depth</span><span class="p">)</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="n">signal</span> <span class="o">=</span> <span class="n">api</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="n">query</span><span class="p">)</span>
    <span class="k">except</span><span class="p">:</span>
        <span class="k">return</span> <span class="kc">None</span><span class="p">,</span> <span class="kc">None</span><span class="p">,</span> <span class="kc">None</span><span class="p">,</span> <span class="kc">None</span>

    <span class="n">outer</span><span class="p">,</span> <span class="n">inner</span> <span class="o">=</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">2</span>
    <span class="n">query</span> <span class="o">=</span> <span class="s2">&quot;SELECT AVG(</span><span class="si">%s</span><span class="s2">) AS </span><span class="si">%s</span><span class="s2">, STDEV(</span><span class="si">%s</span><span class="s2">) AS </span><span class="si">%s</span><span class="s2"> FROM </span><span class="si">%s</span><span class="s2"> &quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">variable</span><span class="p">,</span> <span class="n">variable</span><span class="o">+</span><span class="s1">&#39;_bkg&#39;</span><span class="p">,</span> <span class="n">variable</span><span class="p">,</span> <span class="n">variable</span><span class="o">+</span><span class="s1">&#39;_bkg_std&#39;</span><span class="p">,</span> <span class="n">table</span><span class="p">)</span>
    <span class="n">query</span> <span class="o">+=</span> <span class="s2">&quot;WHERE [time] BETWEEN &#39;</span><span class="si">%s</span><span class="s2">&#39; AND &#39;</span><span class="si">%s</span><span class="s2">&#39; AND &quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">dt1</span><span class="p">,</span> <span class="n">dt2</span><span class="p">)</span>
    <span class="n">query</span> <span class="o">+=</span> <span class="s2">&quot;[lat] BETWEEN </span><span class="si">%f</span><span class="s2"> AND </span><span class="si">%f</span><span class="s2"> AND &quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">lat</span><span class="o">-</span><span class="n">outer</span><span class="o">*</span><span class="n">del_lat</span><span class="p">,</span> <span class="n">lat</span><span class="o">+</span><span class="n">outer</span><span class="o">*</span><span class="n">del_lat</span><span class="p">)</span>
    <span class="n">query</span> <span class="o">+=</span> <span class="s2">&quot;[lat] NOT BETWEEN </span><span class="si">%f</span><span class="s2"> AND </span><span class="si">%f</span><span class="s2"> AND &quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">lat</span><span class="o">-</span><span class="n">inner</span><span class="o">*</span><span class="n">del_lat</span><span class="p">,</span> <span class="n">lat</span><span class="o">+</span><span class="n">inner</span><span class="o">*</span><span class="n">del_lat</span><span class="p">)</span>
    <span class="n">query</span> <span class="o">+=</span> <span class="s2">&quot;[lon] BETWEEN </span><span class="si">%f</span><span class="s2"> AND </span><span class="si">%f</span><span class="s2"> AND &quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">lon</span><span class="o">-</span><span class="n">outer</span><span class="o">*</span><span class="n">del_lon</span><span class="p">,</span> <span class="n">lon</span><span class="o">+</span><span class="n">outer</span><span class="o">*</span><span class="n">del_lon</span><span class="p">)</span>
    <span class="n">query</span> <span class="o">+=</span> <span class="s2">&quot;[lon] NOT BETWEEN </span><span class="si">%f</span><span class="s2"> AND </span><span class="si">%f</span><span class="s2"> &quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">lon</span><span class="o">-</span><span class="n">inner</span><span class="o">*</span><span class="n">del_lon</span><span class="p">,</span> <span class="n">lon</span><span class="o">+</span><span class="n">inner</span><span class="o">*</span><span class="n">del_lon</span><span class="p">)</span>
    <span class="k">if</span> <span class="n">has_depth</span><span class="p">(</span><span class="n">table</span><span class="p">):</span>
        <span class="n">query</span> <span class="o">+=</span> <span class="s2">&quot;AND [depth] BETWEEN </span><span class="si">%f</span><span class="s2"> AND </span><span class="si">%f</span><span class="s2"> &quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">depth</span><span class="o">-</span><span class="n">del_depth</span><span class="p">,</span> <span class="n">depth</span><span class="o">+</span><span class="n">del_depth</span><span class="p">)</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="n">background</span> <span class="o">=</span> <span class="n">api</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="n">query</span><span class="p">)</span>
    <span class="k">except</span><span class="p">:</span>
        <span class="k">return</span> <span class="kc">None</span><span class="p">,</span> <span class="kc">None</span><span class="p">,</span> <span class="kc">None</span><span class="p">,</span> <span class="kc">None</span>
    <span class="n">sig</span><span class="p">,</span> <span class="n">sig_bkg</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span> <span class="kc">None</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">signal</span><span class="p">)</span><span class="o">&gt;</span><span class="mi">0</span><span class="p">:</span> <span class="n">sig</span><span class="p">,</span> <span class="n">sig_bkg</span> <span class="o">=</span> <span class="n">signal</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="n">variable</span><span class="p">],</span> <span class="n">signal</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="n">variable</span><span class="o">+</span><span class="s1">&#39;_std&#39;</span><span class="p">]</span>
    <span class="k">except</span><span class="p">:</span>
        <span class="n">sig</span><span class="p">,</span> <span class="n">sig_bkg</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span> <span class="kc">None</span>
    <span class="n">bkg</span><span class="p">,</span> <span class="n">bkg_std</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span> <span class="kc">None</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">background</span><span class="p">)</span><span class="o">&gt;</span><span class="mi">0</span><span class="p">:</span> <span class="n">bkg</span><span class="p">,</span> <span class="n">bkg_std</span> <span class="o">=</span> <span class="n">background</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="n">variable</span><span class="o">+</span><span class="s1">&#39;_bkg&#39;</span><span class="p">],</span> <span class="n">background</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="mi">0</span><span class="p">,</span> <span class="n">variable</span><span class="o">+</span><span class="s1">&#39;_bkg_std&#39;</span><span class="p">]</span>
    <span class="k">except</span><span class="p">:</span>
        <span class="n">bkg</span><span class="p">,</span> <span class="n">bkg_std</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span> <span class="kc">None</span>
    <span class="k">return</span> <span class="n">sig</span><span class="p">,</span> <span class="n">sig_bkg</span><span class="p">,</span> <span class="n">bkg</span><span class="p">,</span> <span class="n">bkg_std</span>




<span class="k">def</span> <span class="nf">match_params</span><span class="p">():</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Prepares a list variables (and their associated tolerances) to be colocalized with eddies.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">Param</span> <span class="o">=</span> <span class="n">namedtuple</span><span class="p">(</span><span class="s1">&#39;Param&#39;</span><span class="p">,</span> <span class="p">[</span><span class="s1">&#39;table&#39;</span><span class="p">,</span> <span class="s1">&#39;variable&#39;</span><span class="p">,</span> <span class="s1">&#39;temporalTolerance&#39;</span><span class="p">,</span> <span class="s1">&#39;latTolerance&#39;</span><span class="p">,</span> <span class="s1">&#39;lonTolerance&#39;</span><span class="p">,</span> <span class="s1">&#39;depthTolerance&#39;</span><span class="p">])</span>
    <span class="n">params</span> <span class="o">=</span> <span class="p">[]</span>

    <span class="c1">######## satellite</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblSST_AVHRR_OI_NRT&#39;</span><span class="p">,</span> <span class="s1">&#39;sst&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
<span class="c1">#     params.append(Param(&#39;tblSSS_NRT&#39;, &#39;sss&#39;, 0, 0.5, 0.5, 5))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblCHL_REP&#39;</span><span class="p">,</span> <span class="s1">&#39;chl&#39;</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
<span class="c1">#     params.append(Param(&#39;tblModis_AOD_REP&#39;, &#39;AOD&#39;, 15, 1, 1, 5))</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblAltimetry_REP&#39;</span><span class="p">,</span> <span class="s1">&#39;sla&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>

    <span class="c1">######## model</span>
    <span class="n">params</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">Param</span><span class="p">(</span><span class="s1">&#39;tblPisces_NRT&#39;</span><span class="p">,</span> <span class="s1">&#39;NO3&#39;</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mi">5</span><span class="p">))</span>
<span class="c1">#     params.append(Param(&#39;tblPisces_NRT&#39;, &#39;Fe&#39;, 4, 0.5, 0.5, 5))</span>
<span class="c1">#     params.append(Param(&#39;tblPisces_NRT&#39;, &#39;O2&#39;, 4, 0.5, 0.5, 5))</span>
<span class="c1">#     params.append(Param(&#39;tblPisces_NRT&#39;, &#39;PO4&#39;, 4, 0.5, 0.5, 5))</span>
<span class="c1">#     params.append(Param(&#39;tblPisces_NRT&#39;, &#39;Si&#39;, 4, 0.5, 0.5, 5))</span>
<span class="c1">#     params.append(Param(&#39;tblPisces_NRT&#39;, &#39;PP&#39;, 4, 0.5, 0.5, 5))</span>
<span class="c1">#     params.append(Param(&#39;tblPisces_NRT&#39;, &#39;CHL&#39;, 4, 0.5, 0.5, 5))</span>
<span class="c1">#     params.append(Param(&#39;tblDarwin_Nutrient&#39;, &#39;PO4&#39;, 2, 0.5, 0.5, 5))</span>
<span class="c1">#     params.append(Param(&#39;tblDarwin_Nutrient&#39;, &#39;SiO2&#39;, 2, 0.5, 0.5, 5))</span>
<span class="c1">#     params.append(Param(&#39;tblDarwin_Nutrient&#39;, &#39;O2&#39;, 2, 0.5, 0.5, 5))</span>
<span class="c1">#     params.append(Param(&#39;tblDarwin_Ecosystem&#39;, &#39;phytoplankton&#39;, 2, 0.5, 0.5, 5))</span>
<span class="c1">#     params.append(Param(&#39;tblDarwin_Ecosystem&#39;, &#39;zooplankton&#39;, 2, 0.5, 0.5, 5))</span>
<span class="c1">#     params.append(Param(&#39;tblDarwin_Ecosystem&#39;, &#39;CHL&#39;, 2, 0.5, 0.5, 5))</span>
<span class="c1">#     params.append(Param(&#39;tblDarwin_Ecosystem&#39;, &#39;primary_production&#39;, 2, 0.5, 0.5, 5))</span>
<span class="c1">#     params.append(Param(&#39;tblDarwin_Phytoplankton&#39;, &#39;diatom&#39;, 2, 0.5, 0.5, 5))</span>
<span class="c1">#     params.append(Param(&#39;tblDarwin_Phytoplankton&#39;, &#39;coccolithophore&#39;, 2, 0.5, 0.5, 5))</span>
<span class="c1">#     params.append(Param(&#39;tblDarwin_Phytoplankton&#39;, &#39;picoeukaryote&#39;, 2, 0.5, 0.5, 5))</span>
<span class="c1">#     params.append(Param(&#39;tblDarwin_Phytoplankton&#39;, &#39;picoprokaryote&#39;, 2, 0.5, 0.5, 5))</span>
<span class="c1">#     params.append(Param(&#39;tblDarwin_Phytoplankton&#39;, &#39;mixotrophic_dinoflagellate&#39;, 2, 0.5, 0.5, 5))</span>


    <span class="n">tables</span><span class="p">,</span> <span class="n">variables</span><span class="p">,</span> <span class="n">temporalTolerance</span><span class="p">,</span> <span class="n">latTolerance</span><span class="p">,</span> <span class="n">lonTolerance</span><span class="p">,</span> <span class="n">depthTolerance</span> <span class="o">=</span> <span class="p">[],</span> <span class="p">[],</span> <span class="p">[],</span> <span class="p">[],</span> <span class="p">[],</span> <span class="p">[]</span>
    <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">params</span><span class="p">)):</span>
        <span class="n">tables</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">params</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">table</span><span class="p">)</span>
        <span class="n">variables</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">params</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">variable</span><span class="p">)</span>
        <span class="n">temporalTolerance</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">params</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">temporalTolerance</span><span class="p">)</span>
        <span class="n">latTolerance</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">params</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">latTolerance</span><span class="p">)</span>
        <span class="n">lonTolerance</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">params</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">lonTolerance</span><span class="p">)</span>
        <span class="n">depthTolerance</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">params</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">.</span><span class="n">depthTolerance</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">tables</span><span class="p">,</span> <span class="n">variables</span><span class="p">,</span> <span class="n">temporalTolerance</span><span class="p">,</span> <span class="n">latTolerance</span><span class="p">,</span> <span class="n">lonTolerance</span><span class="p">,</span> <span class="n">depthTolerance</span>



<span class="k">def</span> <span class="nf">main</span><span class="p">(</span><span class="n">y1</span><span class="p">,</span> <span class="n">y2</span><span class="p">,</span> <span class="n">m1</span><span class="p">,</span> <span class="n">m2</span><span class="p">,</span> <span class="n">d1</span><span class="p">,</span> <span class="n">d2</span><span class="p">,</span> <span class="n">edd_lat1</span><span class="p">,</span> <span class="n">edd_lat2</span><span class="p">,</span> <span class="n">edd_lon1</span><span class="p">,</span> <span class="n">edd_lon2</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Instantiates the API class and using the &#39;match_covariate()&#39; function colocalizes the retrieved eddies</span>
<span class="sd">    with the specified variables.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">api</span> <span class="o">=</span> <span class="n">pycmap</span><span class="o">.</span><span class="n">API</span><span class="p">()</span>
    <span class="n">daysDir</span> <span class="o">=</span> <span class="s1">&#39;./export/eddy/days/&#39;</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">exists</span><span class="p">(</span><span class="n">daysDir</span><span class="p">):</span> <span class="n">os</span><span class="o">.</span><span class="n">makedirs</span><span class="p">(</span><span class="n">daysDir</span><span class="p">)</span>

    <span class="n">days</span> <span class="o">=</span> <span class="n">sparse_dates</span><span class="p">(</span><span class="n">y1</span><span class="p">,</span> <span class="n">y2</span><span class="p">,</span> <span class="n">m1</span><span class="p">,</span> <span class="n">m2</span><span class="p">,</span> <span class="n">d1</span><span class="p">,</span> <span class="n">d2</span><span class="p">)</span>
    <span class="n">tables</span><span class="p">,</span> <span class="n">variables</span><span class="p">,</span> <span class="n">temporalTolerance</span><span class="p">,</span> <span class="n">latTolerance</span><span class="p">,</span> <span class="n">lonTolerance</span><span class="p">,</span> <span class="n">depthTolerance</span> <span class="o">=</span> <span class="n">match_params</span><span class="p">()</span>

    <span class="k">for</span> <span class="n">day_ind</span><span class="p">,</span> <span class="n">day</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">days</span><span class="p">):</span>
        <span class="n">eddies</span> <span class="o">=</span> <span class="n">daily_eddies</span><span class="p">(</span><span class="n">api</span><span class="p">,</span> <span class="nb">str</span><span class="p">(</span><span class="n">day</span><span class="p">),</span> <span class="n">edd_lat1</span><span class="p">,</span> <span class="n">edd_lat2</span><span class="p">,</span> <span class="n">edd_lon1</span><span class="p">,</span> <span class="n">edd_lon2</span><span class="p">)</span>
        <span class="n">eddies</span><span class="p">[</span><span class="s1">&#39;time&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">to_datetime</span><span class="p">(</span><span class="n">eddies</span><span class="p">[</span><span class="s1">&#39;time&#39;</span><span class="p">])</span>
        <span class="k">for</span> <span class="n">variable</span> <span class="ow">in</span> <span class="n">variables</span><span class="p">:</span>
            <span class="n">eddies</span><span class="p">[</span><span class="n">variable</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span>
            <span class="n">eddies</span><span class="p">[</span><span class="n">variable</span><span class="o">+</span><span class="s1">&#39;_std&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span>
            <span class="n">eddies</span><span class="p">[</span><span class="n">variable</span><span class="o">+</span><span class="s1">&#39;_bkg&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span>
            <span class="n">eddies</span><span class="p">[</span><span class="n">variable</span><span class="o">+</span><span class="s1">&#39;_bkg_std&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span>
        <span class="nb">print</span><span class="p">(</span><span class="s1">&#39;Day </span><span class="si">%s</span><span class="s1">:  </span><span class="si">%d</span><span class="s1"> / </span><span class="si">%d</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">day</span><span class="p">),</span> <span class="n">day_ind</span><span class="o">+</span><span class="mi">1</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">days</span><span class="p">)))</span>
        <span class="k">for</span> <span class="n">e</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">eddies</span><span class="p">)):</span>
            <span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">Eddy </span><span class="si">%d</span><span class="s1"> / </span><span class="si">%d</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">e</span><span class="o">+</span><span class="mi">1</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">eddies</span><span class="p">)))</span>
            <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">variables</span><span class="p">)):</span>
                <span class="c1"># print(&#39;\t\t%d. Matching %s&#39; % (i+1, variables[i]))</span>
                <span class="n">dt1</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">eddies</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">e</span><span class="p">,</span> <span class="s1">&#39;time&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="n">timedelta</span><span class="p">(</span><span class="n">days</span><span class="o">=-</span><span class="n">temporalTolerance</span><span class="p">[</span><span class="n">i</span><span class="p">]))</span>
                <span class="n">dt2</span> <span class="o">=</span> <span class="nb">str</span><span class="p">(</span><span class="n">eddies</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">e</span><span class="p">,</span> <span class="s1">&#39;time&#39;</span><span class="p">]</span> <span class="o">+</span> <span class="n">timedelta</span><span class="p">(</span><span class="n">days</span><span class="o">=</span><span class="n">temporalTolerance</span><span class="p">[</span><span class="n">i</span><span class="p">]))</span>
                <span class="n">lat</span><span class="p">,</span> <span class="n">del_lat</span> <span class="o">=</span> <span class="n">eddies</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">e</span><span class="p">,</span> <span class="s1">&#39;lat&#39;</span><span class="p">],</span> <span class="n">latTolerance</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
                <span class="n">lon</span><span class="p">,</span> <span class="n">del_lon</span> <span class="o">=</span> <span class="n">eddies</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">e</span><span class="p">,</span> <span class="s1">&#39;lon&#39;</span><span class="p">],</span> <span class="n">lonTolerance</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
                <span class="n">depth</span><span class="p">,</span> <span class="n">del_depth</span> <span class="o">=</span> <span class="mi">0</span><span class="p">,</span> <span class="n">depthTolerance</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
                <span class="n">v</span><span class="p">,</span> <span class="n">v_std</span><span class="p">,</span> <span class="n">bkg</span><span class="p">,</span> <span class="n">bkg_std</span> <span class="o">=</span> <span class="n">match_covariate</span><span class="p">(</span><span class="n">api</span><span class="p">,</span> <span class="n">tables</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="n">variables</span><span class="p">[</span><span class="n">i</span><span class="p">],</span> <span class="n">dt1</span><span class="p">,</span> <span class="n">dt2</span><span class="p">,</span> <span class="n">lat</span><span class="p">,</span> <span class="n">del_lat</span><span class="p">,</span> <span class="n">lon</span><span class="p">,</span> <span class="n">del_lon</span><span class="p">,</span> <span class="n">depth</span><span class="p">,</span> <span class="n">del_depth</span><span class="p">)</span>
                <span class="n">eddies</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">e</span><span class="p">,</span> <span class="n">variables</span><span class="p">[</span><span class="n">i</span><span class="p">]]</span> <span class="o">=</span> <span class="n">v</span>
                <span class="n">eddies</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">e</span><span class="p">,</span> <span class="n">variables</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">+</span><span class="s1">&#39;_std&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">v_std</span>
                <span class="n">eddies</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">e</span><span class="p">,</span> <span class="n">variables</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">+</span><span class="s1">&#39;_bkg&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">bkg</span>
                <span class="n">eddies</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">e</span><span class="p">,</span> <span class="n">variables</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">+</span><span class="s1">&#39;_bkg_std&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">bkg_std</span>
        <span class="n">eddies</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="n">daysDir</span><span class="o">+</span><span class="nb">str</span><span class="p">(</span><span class="n">day</span><span class="o">.</span><span class="n">date</span><span class="p">())</span><span class="o">+</span><span class="s1">&#39;.csv&#39;</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">eddies</span>







<span class="c1">##############################</span>
<span class="c1">#                            #</span>
<span class="c1">#           main             #</span>
<span class="c1">#                            #</span>
<span class="c1">##############################</span>




<span class="k">if</span> <span class="vm">__name__</span> <span class="o">==</span> <span class="s1">&#39;__main__&#39;</span><span class="p">:</span>
    <span class="c1">### time window</span>
    <span class="n">y1</span><span class="p">,</span> <span class="n">y2</span> <span class="o">=</span> <span class="mi">2014</span><span class="p">,</span> <span class="mi">2014</span>
    <span class="n">m1</span><span class="p">,</span> <span class="n">m2</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">1</span>
    <span class="n">d1</span><span class="p">,</span> <span class="n">d2</span> <span class="o">=</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">1</span>
    <span class="c1">### spatial range</span>
    <span class="n">edd_lat1</span><span class="p">,</span> <span class="n">edd_lat2</span> <span class="o">=</span> <span class="mi">20</span><span class="p">,</span> <span class="mi">30</span>
    <span class="n">edd_lon1</span><span class="p">,</span> <span class="n">edd_lon2</span> <span class="o">=</span> <span class="o">-</span><span class="mi">160</span><span class="p">,</span> <span class="o">-</span><span class="mi">150</span>
    <span class="n">eddies</span> <span class="o">=</span> <span class="n">main</span><span class="p">(</span><span class="n">y1</span><span class="p">,</span> <span class="n">y2</span><span class="p">,</span> <span class="n">m1</span><span class="p">,</span> <span class="n">m2</span><span class="p">,</span> <span class="n">d1</span><span class="p">,</span> <span class="n">d2</span><span class="p">,</span> <span class="n">edd_lat1</span><span class="p">,</span> <span class="n">edd_lat2</span><span class="p">,</span> <span class="n">edd_lon1</span><span class="p">,</span> <span class="n">edd_lon2</span><span class="p">)</span>

</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Day 2014-01-01 00:00:00:  1 / 1
        Eddy 1 / 12
        Eddy 2 / 12
        Eddy 3 / 12
        Eddy 4 / 12
        Eddy 5 / 12
        Eddy 6 / 12
        Eddy 7 / 12
        Eddy 8 / 12
        Eddy 9 / 12
        Eddy 10 / 12
        Eddy 11 / 12
        Eddy 12 / 12
</pre></div></div>
</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span>
</pre></div>
</div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Mohammad Dehghani Ashkezari &amp; Norland Raphael Hagen
      <span class="lastupdated">
        Last updated on Dec 31, 2019.
      </span>

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>